doc: Add release notes for 3.0.0 (#21339)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-3.0.0.rst

  * Erasing the external memory when programming a new firmware image with ``west flash`` series now always correctly honors the ``--erase`` flag (and its absence) both when using the ``nrfjprog`` and ``nrfutil`` backends.
    Prior to this release, the ``nrjfprog`` backend would always erase only the sectors of the external flash used by the new firmware, and the ``nrfutil`` backend would always erase the whole external flash.

Developing with nRF70 Series
============================

* Added

  * Support for the nRF7002 EB II (PCA63571) with the nRF54 Series DKs as detailed in :ref:`ug_nrf7002eb2_gs`.
  * A new section :ref:`ug_nrf70_wifi_enterprise_mode` in the :ref:`ug_nrf70_wifi_advanced_security_modes` page.
* Deprecated support for the nRF7002 EB (PCA63561) with the nRF54 Series DKs.
* Removed support for storing the nRF70 firmware patches in external flash without the :ref:`partition_manager`, as mentioned in :ref:`ug_nrf70_developing_fw_patch_ext_flash`.

Developing with nRF54L Series
=============================

* Added HMAC SHA-256 with a 128-bit key type to KMU, as detailed in the :ref:`ug_nrf54l_crypto_kmu_supported_key_types` documentation section.

> 

Reviewer: > Added workaround for nRF54L15 Errata 30. Use `CLOCK_CONTROL_NRF_HF_CALIBRATION=y` symbol to explicitly activate the workaround. Consequence of activating the workaround might be increased power consumption due to periodic CPU wake-up, so use it only if errata conditions are met. Note that the workaround is already included in MPSL component, so the solution is applicable needed only if your application does not utilize MPSL.

Once https://github.com/nrfconnect/sdk-nrf/pull/21391 is merged (and backported to v3.0 branch)

Author: Added.

Matter fork
+++++++++++

* Added a new ``kFactoryReset`` event that is posted during factory reset.
  The application can register a handler and perform additional cleanup.

Thread
------

* Added:

  * Support for storing the Thread key materials in the :ref:`ug_ot_thread_security_kmu`.
  * The :ref:`ug_ot_thread_security` user guide describing the security features of the |NCS| OpenThread implementation.

Zigbee
------

* Removed all Zigbee resources.
  They are now available as separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories.

Wi-Fi®
------

* The :ref:`ug_wifi_regulatory_certification` documentation is now moved under :ref:`ug_wifi` protocol page.

Applications
============

* Removed the Asset Tracker v2 application.
  For development of asset tracking applications, refer to the `Asset Tracker Template <Asset Tracker Template_>`_.

  The factory-programmed Asset Tracker v2 firmware is still available to program the nRF91 Series devices using the `Programmer app`_, `Quick Start app`_ , and the `Cellular Monitor app`_.

Machine learning
----------------

> * Updated the application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system for the :ref:`zephyr:nrf54h20dk_nrf54h20` board.

Reviewer 2: ```suggestion
* Updated the application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system for the :ref:`zephyr:nrf54h20dk_nrf54h20` board.
* Update the Edge Impulse URI configuration to use the new model location.
```

After this one: #21423 is merged.

Author: Added

  * Removed the Running applications with Trusted Firmware-M page.
    Its contents have been moved into the following new pages:

    * :ref:`ug_tfm_index`
    * :ref:`ug_tfm_building`
    * :ref:`ug_tfm_logging`
    * :ref:`ug_tfm_services`
    * :ref:`ug_tfm_provisioning`

Protocols
=========

Bluetooth® LE
-------------

* Updated the Bluetooth LE SoftDevice Controller driver to make the :c:func:`hci_vs_sdc_llpm_mode_set` function return an error if Low Latency Packet Mode (LLPM) is not supported or not enabled in the Bluetooth LE Controller driver configuration (:kconfig:option:`CONFIG_BT_CTLR_SDC_LLPM`).

* Fixed:

  * An issue where a flash operation executed on the system workqueue might result in ``-ETIMEDOUT``, if there is an active Bluetooth LE connection.
  * An issue where Bluetooth applications built with the ``nordic-bt-rpc`` snippet (in the :ref:`ble_rpc` configuration) did not work on the nRF54H20 devices due to incorrect memory mapping.

* Removed the ``HCI_LE_Read_Local_P-256_Public_Key`` and ``HCI_LE_Generate_DHKey`` commands emulation from HCI driver.

Bluetooth Mesh
--------------

> * Added the key importer functionality (:kconfig:option:`CONFIG_BT_MESH_KEY_IMPORTER`).

Reviewer 3: Please add:
```
* Added a note to the :ref:`dfu_over_ble` page about a need to disable application settings erase option in the `nRF Connect for Mobile`_ and `nRF Connect Device Manager`_ mobile applications when performing P2P FOTA over Bluetooth Low Energy.
```

Author: Added,

    The feature is enabled by default if Zephyr power management (:kconfig:option:`CONFIG_PM`) is enabled.
    It prevents entering power states that introduce wakeup latency and ensures high performance.
  * Static Partition Manager memory maps for single-image configurations (without bootloader and separate radio/network core image).
    In the |NCS|, the Partition Manager is enabled by default for single-image sysbuild builds.
    The static memory map ensures control over settings partition placement and size.
    The introduced static memory maps might not be consistent with the ``storage_partition`` defined by the board-level DTS configuration.
  * Support for GATT long (reliable) writes (:kconfig:option:`CONFIG_BT_ATT_PREPARE_COUNT`) to Fast Pair and Works With ChromeBook (WWCB) configurations.
    This allows performing :ref:`fwupd <nrf_desktop_fwupd>` DFU image upload over Bluetooth LE with GATT clients that do not perform MTU exchange (for example, ChromeOS using the Floss Bluetooth stack).
  * The ``dongle`` and ``release_dongle`` application configurations for the nRF54H20 Development Kit (``nrf54h20dk/nrf54h20/cpuapp``).
    The configurations act as a HID dongle.

> * Updated:

Reviewer 4: Please add under nRF Desktop updated (assuming https://github.com/nrfconnect/sdk-nrf/pull/21466 goes in):
```
  * Disabled the UDC DWC2 DMA support (:kconfig:option:`CONFIG_UDC_DWC2_DMA`) on nRF54H20 DK.
    The DMA support is experimental, disabling the feature improves USB HID stability.
    Since nRF Desktop uses only small HID reports (report size is smaller than 64 bytes), the DMA doesn't improve performance.
```

Author: Added.

      This change ensures consistency with the sample release configuration that has the LTO feature enabled by default.
    * The ``nrf54l15dk/nrf54l15/cpuapp`` board target configuration to enable hardware cryptography for the MCUboot bootloader.
      The application image is verified using a pure ED25519 signature and the public key used by MCUboot for validating the application image is securely stored in the Key Management Unit (KMU) hardware peripheral.
      Support for the ``nrf54l15dk/nrf54l05/cpuapp`` and ``nrf54l15dk/nrf54l10/cpuapp`` board targets, which is added to this sample in this release iteration, also includes the same MCUboot bootloader configuration with the hardware cryptography enabled.

      The change modifies the memory partition layout for the ``nrf54l15dk/nrf54l15/cpuapp`` board target and changes the MCUboot image signing algorithm.
      Because of that, the application images built for the ``nrf54l15dk/nrf54l15/cpuapp`` board target from this |NCS| release are not compatible with the MCUboot bootloader built from previous releases.
      It is highly recommended to use hardware cryptography for the nRF54L SoC Series for improved security.

>     * The configurations for board targets with the MCUboot bootloader support to use a non-default signature key file (the ``SB_CONFIG_BOOT_SIGNATURE_KEY_FILE`` Kconfig option).

Reviewer 5: Please add new entry under `* :ref:fast_pair_locator_tag sample: * Updated:` section:
```
* The MCUBoot DFU signature type to the Elliptic curve digital signatures with curve P-256 (ECDSA P256 - the ``SB_CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256`` Kconfig option) for the ``nrf52840dk/nrf52840`` board target.
  This is done to use Cryptocell 310 for image signature verification.
  This change breaks the backwards compatibility, as performing DFU from an old signature type to a new one is impossible.
```

Reviewer 5: @kapi-no, @peknis please review

Reviewer 6: Looks good

Author: Added

* Added support for the ``nrf54l15dk/nrf54l05/cpuapp`` and ``nrf54l15dk/nrf54l10/cpuapp`` board targets in all Edge Impulse samples.

Enhanced ShockBurst samples
---------------------------

* Added support for the ``nrf54l15dk/nrf54l05/cpuapp`` and ``nrf54l15dk/nrf54l10/cpuapp`` board targets in all ESB samples.

Matter samples
--------------

* Added :ref:`matter_manufacturer_specific_sample` sample that demonstrates an implementation of custom manufacturer-specific clusters used by the application layer.

* :ref:`matter_template_sample` sample:

  * Updated:

    * The documentation with instructions on how to build the sample on the nRF54L15 DK with support for Matter OTA DFU and DFU over Bluetooth SMP, and using internal RRAM only.
    * Link Time Optimization (LTO) to be enabled by default for the ``release`` configuration and ``nrf7002dk/nrf5340/cpuapp`` build target.

  * Removed support for nRF54H20 devices.

* :ref:`matter_lock_sample` sample:

  * Removed support for nRF54H20 devices.
  * Updated the API of ``AppTask``, ``BoltLockManager``, and ``AccessManager`` to provide additional information for the ``LockOperation`` event.

Networking samples
------------------

> 

Reviewer 7: Related to https://github.com/nrfconnect/sdk-nrf/pull/21392.

> 

Author: Added

     * By extending the manifests to process the ``suit-payload-fetch`` sequence of the Nordic top update candidate.
     * By extending the manifests with build-time checks for consistency between MPI and envelope signing configuration.

Trusted Firmware-M (TF-M) samples
---------------------------------

* :ref:`tfm_psa_template` sample:

  * Added support for the following attestation token fields:

    * Profile definition
    * PSA certificate reference (optional), configured using the ``SB_CONFIG_TFM_OTP_PSA_CERTIFICATE_REFERENCE`` sysbuild Kconfig option
    * Verification service URL (optional), configured using the :kconfig:option:`CONFIG_TFM_ATTEST_VERIFICATION_SERVICE_URL` Kconfig option

* :ref:`tfm_secure_peripheral_partition` sample:

  * Updated documentation with information about how to access other TF-M partitions from the secure partition.

Thread samples
--------------

* :ref:`ot_cli_sample` sample:

  * Removed support for the nRF54H20 DK.

Zigbee samples
--------------

* Removed all Zigbee samples.
  They are now available as separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories.

Wi-Fi samples
-------------

> 

Reviewer 8: * Added: 
 
  * Added support for ``nrf7002eb2`` with nRF54 Series devices in the following samples:
 
    * :ref:`wifi_station_sample`
    * :ref:`wifi_scan_sample`
    * :ref:`wifi_shell_sample`
    * :ref:`wifi_radio_test_sd`
    * :ref:`wifi_radio_test`
    * :ref:`wifi_softap_sample`
    * :ref:`wifi_shutdown_sample`
    * :ref:`wifi_twt_sample`
    * :ref:`wifi_offloaded_raw_tx_packet_sample`
    * :ref:`wifi_raw_tx_packet_sample`
    * :ref:`_ble_wifi_provision`

Reviewer 7: Related to https://github.com/nrfconnect/sdk-nrf/pull/21392.

Author: Added

Other samples
-------------

* :ref:`coremark_sample` sample:

  * Added:

    * Support for the nRF54L05 and nRF54L10 SoCs (emulated on nRF54L15 DK).
    * FLPR core support for the :ref:`zephyr:nrf54l15dk_nrf54l15` and :ref:`zephyr:nrf54h20dk_nrf54h20` board targets.

  * Removed the following compiler options that were set in the :kconfig:option:`CONFIG_COMPILER_OPT` Kconfig option:

    * ``-fno-pie``
    * ``-fno-pic``
    * ``-ffunction-sections``
    * ``-fdata-sections``

    These options are enabled by default in Zephyr and do not need to be set with the dedicated Kconfig option.

* :ref:`caf_sensor_manager_sample` sample:

  * Added low power configuration for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target.

Drivers
=======

This section provides detailed lists of changes by :ref:`driver <drivers>`.

* Added a :ref:`flash_ipuc` that allows to manage SUIT IPUC memory through the Zephyr flash API.

Wi-Fi drivers
-------------

* Added:

  * An advanced debug shell for reading and writing registers/memory of the nRF70 Series chip.
    The debug shell can be enabled using the :kconfig:option:`CONFIG_WIFI_NRF70_DEBUG_SHELL` Kconfig option.

> 

Author: Added.

  * Updated the ``wifi_util`` shell command, which is now renamed to ``nrf70 util``.

> Libraries

Reviewer 9: @umapraseeda ZMS backend PR is merged now

> Libraries

Author: Added under the 'Additions specific to NCS' section under Zephyr. Please check if that is correct.

> Libraries

Author: @rghaddab

    * The behavior of the :c:member:`bt_fast_pair_fmdn_info_cb.provisioning_state_changed` callback.
      The callback no longer reports the initial provisioning state after the Fast Pair subsystem is enabled with the :c:func:`bt_fast_pair_enable` function call.
      See the :ref:`migration guide <migration_3.0_recommended>` for mandatory changes and the :ref:`ug_bt_fast_pair_gatt_service_fmdn_info_callbacks_provisioning_state` section in the Fast Pair integration guide for the description on how to track the FMDN provisioning state with the new approach.

  * Removed the sysbuild control over the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is defined in the main (default) image.
    Sysbuild no longer sets the value of this Kconfig option.

* :ref:`bt_mesh` library:

  * Fixed an issue in the :ref:`bt_mesh_light_ctrl_srv_readme` model to automatically resume the Lightness Controller after recalling a scene (NCSDK-30033 known issue).

Common Application Framework
----------------------------

* :ref:`caf_buttons`:

  * Added the possibility of using more GPIOs.
    Earlier, only **GPIO0** and **GPIO1** devices were supported.

>     Now, the generic solution supports all GPIOs available in the DTS.

Reviewer 4: Please add under `caf_buttons` (related to https://github.com/nrfconnect/sdk-nrf/pull/21482):
```
* Added :c:struct:`power_off_event` handling to prevent entering system off state with GPIO interrupt disabled.
  Entering system off state with GPIO interrupt disabled would make CAF Buttons unable to trigger wakeup from the system off state on button press.
```

Author: Added

* :ref:`caf_power_manager`:

>   * Updated:

Reviewer 4: Please add under `caf_power_manager` -> Updated:

```
  * The module implementation to use :c:func:`sys_poweroff` API instead of :c:func:`pm_state_force` API to enter the system off state.
  * The module implementaion to integrate the :c:struct:`power_off_event`.
```

Author: Added.

Build and configuration system
==============================

* Removed support for the deprecated multi-image builds (parent-child images) functionality.
  All |NCS| projects must now use :ref:`sysbuild`.
  See :ref:`child_parent_to_sysbuild_migration` for an overview of differences with parent-child image and how to migrate.
* Updated:

  * The default runner for the ``west flash`` command to `nRF Util`_ instead of ``nrfjprog`` that is part of the archived `nRF Command Line Tools`_.
    For more information, see the :ref:`build system section in the v3.0.0 migration guide <migration_3.0_recommended>` and the :ref:`programming_selecting_runner` section on the programming page.
  * Erasing the external memory when programming a new firmware image with ``west flash`` series now always correctly honors the ``--erase`` flag (and its absence) both when using the ``nrfjprog`` and ``nrfutil`` backends.
    Prior to this release, the ``nrjfprog`` backend would always erase only the sectors of the external flash used by the new firmware, and the ``nrfutil`` backend would always erase the whole external flash.

> 

Author: Added

Developing with nRF70 Series
============================

* Added:

  * Support for the nRF7002 EB II (PCA63571) with the nRF54 Series DKs as detailed in :ref:`ug_nrf7002eb2_gs`.
  * A new section :ref:`ug_nrf70_wifi_enterprise_mode` in the :ref:`ug_nrf70_wifi_advanced_security_modes` page.

* Deprecated support for the nRF7002 EB (PCA63561) with the nRF54 Series DKs.
* Removed support for storing the nRF70 firmware patches in external flash without the :ref:`partition_manager`, as mentioned in :ref:`ug_nrf70_developing_fw_patch_ext_flash`.

Developing with nRF54L Series
=============================

* Added:

  * HMAC SHA-256 with a 128-bit key type to KMU, as detailed in the :ref:`ug_nrf54l_crypto_kmu_supported_key_types` documentation section.
  * A workaround for nRF54L15 Errata 30.
    Use ``CLOCK_CONTROL_NRF_HF_CALIBRATION=y`` to explicitly activate the workaround.
    Consequence of activating the workaround might be increased power consumption due to periodic CPU wake-up, so use it only if errata conditions are met.

>     Note that the workaround is already included in MPSL component, so the solution is applicable needed only if your application does not utilize MPSL.

Reviewer: > so the solution is applicable needed only if your application does not utilize MPSL.

Either `applicable` or `needed` should be probably removed. I guess `applicable` is more appropriate

Author: Fixed

* Added:

  * A new documentation page :ref:`ug_matter_group_communication` in the :ref:`ug_matter_intro_overview`.
  * A new page on :ref:`ug_matter_creating_custom_cluster`.
  * A description for the new :ref:`ug_matter_gs_tools_matter_west_commands_append` within the :ref:`ug_matter_gs_tools_matter_west_commands` page.
  * New arguments to the :ref:`ug_matter_gs_tools_matter_west_commands_zap_tool_gui` to provide a custom cache directory and add new clusters to Matter Data Model.
  * :ref:`ug_matter_debug_snippet`.
  * Storing Matter key materials in the :ref:`matter_platforms_security_kmu`.
  * A new section :ref:`ug_matter_device_low_power_calibration_period` in the :ref:`ug_matter_device_low_power_configuration` page.
  * A new section :ref:`ug_matter_gs_tools_opp` in the :ref:`ug_matter_gs_tools` page.

* Updated:

  * By disabling the :ref:`mpsl` before performing factory reset to speed up the process.
  * The :ref:`ug_matter_device_low_power_configuration` page to mention the `nWP049 - Matter over Thread: Power consumption and battery life`_ and `Online Power Profiler for Matter over Thread`_ as useful resources in optimizing the power consumption of a Matter device.

> 

Author: Added [here](https://github.com/nrfconnect/sdk-nrf/pull/21339/files#diff-fe5d2ad1e3e8cfa1ecd541e8cd2725c58b094c7ebec0c08bfc6a8424a5de1d94R252) and [here](https://github.com/nrfconnect/sdk-nrf/pull/21339/files#diff-fe5d2ad1e3e8cfa1ecd541e8cd2725c58b094c7ebec0c08bfc6a8424a5de1d94R291). Please check.

> 

Reviewer 10: Looks good to me, thanks!

  For development of asset tracking applications, refer to the `Asset Tracker Template <Asset Tracker Template_>`_.

  The factory-programmed Asset Tracker v2 firmware is still available to program the nRF91 Series devices using the `Programmer app`_, `Quick Start app`_ , and the `Cellular Monitor app`_.

Machine learning
----------------

* Updated:

  * The application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system for the :ref:`zephyr:nrf54h20dk_nrf54h20` board.
  * The Edge Impulse URI configuration to use the new model location.

IPC radio firmware
------------------

* Updated:

  * The application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system in all devices that contain MRAM, such as the nRF54H Series devices.
  * The documentation of applications and samples that use the IPC radio firmware as a :ref:`companion component <companion_components>` to mention its usage when built with :ref:`configuration_system_overview_sysbuild`.

Matter bridge
-------------

* Updated by enabling Link Time Optimization (LTO) by default for the ``release`` configuration.
* Removed support for the nRF54H20 devices.

> nRF5340 Audio

Reviewer 11: The PR https://github.com/nrfconnect/sdk-nrf/pull/21285 should probably be mentioned here, somehow. But I don't know if this is a highlight or an `Added` or how to word it correctly

Reviewer 12: Suggest: "Added more information on new DNs and QDIDs"

Also with a link to the QDID/DN table for 5340.

Author: Added.

-------------

* Updated:

  * The documentation for :ref:`nrf53_audio_app_building` with cross-links and additional information.
  * The :file:`buildprog.py` script is an app-specific script for building and programming multiple kits and cores with various audio application configurations.
    The script will be deprecated in a future release.
    The audio applications will gradually shift only to using standard tools for building and programming development kits.

>   * The :ref:`nRF5340 Audio application\'s <nrf53_audio_app>` :ref:`script for building and programming <nrf53_audio_app_building_script>` now builds into a directory for each transport, device type, core, and version combination.

Reviewer 11: ```suggestion
  * The :ref:`nRF5340 Audio application\'s <nrf53_audio_app>` :ref:`script for building and programming <nrf53_audio_app_building_script>` now builds into a directory for each transport, device type, core, and version combination.
  * Build system to use overlay files for each of the four applications instead of using :file:`Kconfig.default`.
  * The :file:`buildprog.py` to demand argument ``--transport`` to set either ``unicast`` or ``broadcast``
```
The formatting is probably not correct though...

Author: Added

> * Fixed the static random address for broadcast source and unicast server.

Reviewer 11: ```suggestion
* Fixed: 
  
  * The static random address for broadcast source and unicast server.
  * Time sync issue that occasionally caused a 1ms difference between Left/Right headset
```

Author: Added

    This change results from the Bluetooth subsystem transition to the PSA cryptographic standard.
    The Bluetooth stack can now use the PSA crypto API in the non-secure domain as all necessary TF-M partitions are configured properly.
  * The configuration of all Fast Pair samples by increasing the following thread stack sizes to prevent stack overflows:

    * The system workqueue thread stack (:kconfig:option:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE`).
    * The Bluetooth receiving thread stack (:kconfig:option:`CONFIG_BT_RX_STACK_SIZE`).

    This change results from the Bluetooth subsystem transition to the PSA cryptographic API.
  * The sysbuild configurations in samples to align them with the sysbuild Kconfig changes for controlling the Fast Pair provisioning process.

* Removed using a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) from configurations.
  The MPSL flash synchronization issue (NCSDK-29354 in the :ref:`known_issues`) is fixed.
  The workaround is no longer needed.

* :ref:`fast_pair_locator_tag` sample:

  * Added support for the following:

    * :ref:`zephyr:nrf54h20dk_nrf54h20` board target.

>     * Firmware update intents on the Android platform.
>       Integrated the new connection authentication callback from the FMDN module and the Device Information Service (DIS) to support firmware version read operation over the Firmware Revision characteristic.
>       For further details on the Android intent feature for firmware updates, see the :ref:`ug_bt_fast_pair_provisioning_register_firmware_update_intent` section of the Fast Pair integration guide.

Reviewer 6: Let's account for the latest documentation PR:

https://github.com/nrfconnect/sdk-nrf/pull/21530

```suggestion
    * Firmware update intents on the Android platform.
      Updated the configuration of the default device model in the Google Nearby Console to properly support this feature (refer to the sample documentation for details).
      Integrated the new connection authentication callback from the FMDN module and the Device Information Service (DIS) to support firmware version read operation over the Firmware Revision characteristic.
      Updated the sample documentation with a new section that contains the description of this feature and a new testing procedure that demonstrates how this feature works.
      For further details on the Android intent feature for firmware updates, see the :ref:`ug_bt_fast_pair_provisioning_register_firmware_update_intent` section of the Fast Pair integration guide.
```

Author: Added

* Radio test samples:

  * Added:

    * The :ref:`wifi_radio_test_sd` sample to demonstrate the Wi-Fi and Bluetooth LE radio test running on the application core.
    * The ``wifi_radio_test get_voltage`` command to read battery voltage.

  * Updated:

    * The :ref:`wifi_radio_test` sample is now moved to :file:`samples/wifi/radio_test/multi_domain`.

* :ref:`wifi_shell_sample` sample:

  * Updated by modifying support for storing the nRF70 firmware patches in external flash using the :ref:`partition_manager`.

Other samples
-------------

* :ref:`coremark_sample` sample:

  * Added:

    * Support for the nRF54L05 and nRF54L10 SoCs (emulated on nRF54L15 DK).
    * FLPR core support for the :ref:`zephyr:nrf54l15dk_nrf54l15` and :ref:`zephyr:nrf54h20dk_nrf54h20` board targets.

  * Removed the following compiler options that were set in the :kconfig:option:`CONFIG_COMPILER_OPT` Kconfig option:

    * ``-fno-pie``
    * ``-fno-pic``
    * ``-ffunction-sections``
    * ``-fdata-sections``

    These options are enabled by default in Zephyr and do not need to be set with the dedicated Kconfig option.

* :ref:`caf_sensor_manager_sample` sample:

>   * Added low power configuration for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target.

Reviewer 13: Please add:
```suggestion
  * Changed the name of Software Defined Peripherals (SDP) to High Performance Framework (HPF)
```
@masz-nordic , please check if this is the correct place (Other samples) and if description is ok

Reviewer 13: Discussed offline, should be ok

Author: @magp-nordic, this is listed as an application here: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/applications.html. So IMO we should be listing this under Application section: https://ncsdoc.z6.web.core.windows.net/PR-21339/nrf/releases_and_maturity/releases/release-notes-3.0.0.html#applications. Also, did you mean to remove the entry for CAF Sensor Manager sample: Added low power configuration for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target. in line 866?

Reviewer 13: @umapraseeda ok, my mistake, this should be an application entry. 
And when it comes to removing the previous 866 line: sorry, that was not my intention. I thought I changed the suggestion to not remove the previous one, but apparently, I didn't.

Author: Added under Applications. Please check.

      The ``FP_MODEL_ID`` and ``FP_ANTI_SPOOFING_KEY`` CMake variables are replaced by the corresponding ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig options.
    * The automatically generated ``bt_fast_pair`` partition definition (in the :file:`subsys/partition_manager/pm.yml.bt_fast_pair` file) to work correctly when building with TF-M.
    * The behavior of the :c:member:`bt_fast_pair_fmdn_info_cb.provisioning_state_changed` callback.
       The callback no longer reports the initial provisioning state after the Fast Pair subsystem is enabled with the :c:func:`bt_fast_pair_enable` function call.
       See the :ref:`migration guide <migration_3.0_recommended>` for mandatory changes and the :ref:`ug_bt_fast_pair_gatt_service_fmdn_info_callbacks_provisioning_state` section in the Fast Pair integration guide for the description on how to track the FMDN provisioning state with the new approach.

  * Removed the sysbuild control over the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is defined in the main (default) image.
     Sysbuild no longer sets the value of this Kconfig option.

> 

Reviewer 14: Please add
```suggestion

* :ref:`wifi_prov_readme` library:
 
  * Added a new section :ref:`wifi_provisioning_protocol`.
  
```
It reflects the change of PR #21261 and backport PR #21625

Reviewer 14: cc: @richabp

Author: Added

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``fdeb7350171279d4637c536fcceaad3fbb775392``, with some |NCS| specific additions.

For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

   git log --oneline fdeb735017 ^beb733919d

For the list of |NCS| specific commits, including commits cherry-picked from upstream, run:

.. code-block:: none

   git log --oneline manifest-rev ^fdeb735017

The current |NCS| main branch is based on revision ``fdeb735017`` of Zephyr.

.. note::
   For possible breaking changes and changes between the latest Zephyr release and the current Zephyr version, refer to the :ref:`Zephyr release notes <zephyr_release_notes>`.

Additions specific to |NCS|
---------------------------

|no_changes_yet_note|

zcbor
=====

|no_changes_yet_note|

Trusted Firmware-M
==================

|no_changes_yet_note|

cJSON
=====

|no_changes_yet_note|

Documentation
=============

> * Added:

Reviewer 15: Need an entry here for the LTE guide PR - https://github.com/nrfconnect/sdk-nrf/pull/21280, under Added:

 `* The :ref:`cellular_psm` page under the :ref:`ug_lte` documentation, and the documentation was spilt into subpages.

    The Bluetooth stack can now use the PSA crypto API in the non-secure domain as all necessary TF-M partitions are configured properly.

  * The configurations of the following samples by increasing the main thread stack size (:kconfig:option:`CONFIG_MAIN_STACK_SIZE`) to prevent stack overflows:

    * :ref:`bluetooth_central_hids`
    * :ref:`peripheral_hids_keyboard`
    * :ref:`peripheral_hids_mouse`

    This change results from the Bluetooth subsystem transition to the PSA cryptographic API.

  * The following samples to use LE Secure Connection pairing (:kconfig:option:`CONFIG_BT_SMP_SC_PAIR_ONLY`).

    * :ref:`peripheral_gatt_dm`
    * :ref:`peripheral_mds`
    * :ref:`peripheral_cts_client`

* :ref:`direct_test_mode` sample:

  * Added:

    * Loading of radio trims and a fix of a hardware errata for the nRF54H20 SoC to improve the RF performance.
    * Support for the :ref:`nRF2220 front-end module <ug_radio_fem_nrf2220ek>`.
    * Workaround for the hardware errata HMPAN-216 for the nRF54H20 SoC.

* :ref:`central_uart` sample:

  * Added reconnection to bonded devices based on their address.

> 

Reviewer 16: ```suggestion

* :ref:`power_profiling` sample:

  * Added :kconfig:option:`CONFIG_BT_POWER_PROFILING_NFC_DISABLED` to reduce the power consumption by disabling the NFC.

```

> Bluetooth Mesh samples

Reviewer 17: We have another Bluetooth Mesh samples section further below.

  All |NCS| projects must now use :ref:`sysbuild`.
  See :ref:`child_parent_to_sysbuild_migration` for an overview of differences with parent-child image and how to migrate.
* Updated:

  * The default runner for the ``west flash`` command to `nRF Util`_ instead of ``nrfjprog`` that is part of the archived `nRF Command Line Tools`_.
    For more information, see the :ref:`build system section in the v3.0.0 migration guide <migration_3.0_recommended>` and the :ref:`programming_selecting_runner` section on the programming page.
  * Erasing the external memory when programming a new firmware image with ``west flash`` series now always correctly honors the ``--erase`` flag (and its absence) both when using the ``nrfjprog`` and ``nrfutil`` backends.
    Prior to this release, the ``nrjfprog`` backend would always erase only the sectors of the external flash used by the new firmware, and the ``nrfutil`` backend would always erase the whole external flash.
  * :kconfig:option:`CONFIG_NRF53_MULTI_IMAGE_UPDATE` No longer depends on external flash (NCSIDB-1232).
  * Static partition manager file for network core images can now be set (NCSIDB-1442).

> 

Reviewer 18: Please add:
```suggestion
Bootloader and DFU
=============
* Added experimental support |NSIB| for nRF54L15 SoC. On nRF54L SoCs NSIB uses KMU for authentication key storage, it supports keys revocation scheme and it can be protected using immutable-boot region SoC's hardware feature.
* Improve DFU timing performance on nRF54l SoC by applying optimal RRAMC buffering.
```

Author: Added

> 

Reviewer 18: Please add:
* extend QSPI XIP support to include building with TF-M on nRF5340 device. Instance of TF-M can be now part of the internal NVM application image.

Author: Added

    Consequence of activating the workaround might be increased power consumption due to periodic CPU wake-up, so use it only if errata conditions are met.
    Note that the workaround is already included in MPSL component, so the solution is applicable only if your application does not utilize MPSL.

Developing with nRF54H Series
=============================

* Added a new documentation page :ref:`ug_nrf54h20_suit_signing` under :ref:`ug_nrf54h20_suit_dfu`.
* Removed the note on installing SEGGER USB Driver for J-Link on Windows from the :ref:`ug_nrf54h20_gs` page and moved its contents to the `nRF Util prerequisites`_ documentation.
  The Windows-only requirement to install the SEGGER USB Driver for J-Link is now mentioned in the :ref:`installing_vsc` section on the :ref:`installation` page.

Developing with Front-End Modules
=================================

* Added support for the following:

  * :ref:`nRF2220 Front-End Module <ug_radio_fem_nrf2220>`.
  * :ref:`nRF2220 EK shield <ug_radio_fem_nrf2220ek>`.
  * :ref:`nRF21540 Front-End Module in GPIO mode <ug_radio_fem_nrf21540_gpio>` for the nRF54L Series devices.

> 

Reviewer 19: Please add:

```
* Fixed an issue for the nRF21540 Front-End Module (for GPIO and GPIO+SPI modes) when due to late activation of the ``TX_EN`` pin spurious emission occurred.
  The ``PDN`` pin is now activated earlier by the call to the :c:func:`mpsl_fem_enable` function.
  The ``TX_EN`` pin is now activated 15 us earlier during the ramp-up of the radio.
```

Author: Added

* Fixed:

  * An issue where a flash operation executed on the system workqueue might result in ``-ETIMEDOUT``, if there is an active Bluetooth LE connection.
  * An issue where Bluetooth applications built with the ``nordic-bt-rpc`` snippet (in the :ref:`ble_rpc` configuration) did not work on the nRF54H20 devices due to incorrect memory mapping.

* Removed the ``HCI_LE_Read_Local_P-256_Public_Key`` and ``HCI_LE_Generate_DHKey`` commands emulation from HCI driver.

Bluetooth Mesh
--------------

* Added:

  * The key importer functionality (:kconfig:option:`CONFIG_BT_MESH_KEY_IMPORTER`).
  * A note to the :ref:`dfu_over_ble` page about a need to disable application settings erase option in the `nRF Connect for Mobile`_ and `nRF Connect Device Manager`_ mobile applications when performing P2P FOTA over Bluetooth Low Energy.

* Updated the default value for :kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE` if :kconfig:option:`CONFIG_BT_MESH_NLC_PERF_CONF` is selected.

>   The :kconfig:option:`CONFIG_BT_MESH_NLC_PERF_CONF` Kconfig option increases the number of keys used by Mesh and the value of :kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE` Kconfig option needs to be increased accordingly.

Reviewer 3: I dug through our docs. I think it is better to say `the mesh stack` (not capitalized) also specifies what `mesh` means here. We definitely don't use `Mesh` as a single word anywhere. It is either `Bluetooth Mesh`, `the mesh stack`, `a mesh network`, etc.

>   The :kconfig:option:`CONFIG_BT_MESH_NLC_PERF_CONF` Kconfig option increases the number of keys used by Mesh and the value of :kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE` Kconfig option needs to be increased accordingly.

Author: Updated

> * Deprecation procedure has started for :kconfig:option:`CONFIG_BT_MESH_USES_TINYCYPT` and it is not recommended for future designs.

Reviewer 3: Fine by suggestion, but then there should probably be 2 sentences:
```suggestion
* Deprecation procedure has started for :kconfig:option:`CONFIG_BT_MESH_USES_TINYCYPT`.
  It is not recommended to use this Kconfig option for future designs.
```

> * Deprecation procedure has started for :kconfig:option:`CONFIG_BT_MESH_USES_TINYCYPT` and it is not recommended for future designs.

Reviewer 17: Fully agree with Pavel.

> * Deprecation procedure has started for :kconfig:option:`CONFIG_BT_MESH_USES_TINYCYPT` and it is not recommended for future designs.

Author: @PavelVPV what about 'Deprecated the Kconfig... ' (Divya's suggestion) or do you want to retain the content as is 'Deprecation procedure has started..'

> * Deprecation procedure has started for :kconfig:option:`CONFIG_BT_MESH_USES_TINYCYPT` and it is not recommended for future designs.

Author: Updated

  * A new page on :ref:`ug_matter_creating_custom_cluster`.
  * A description for the new :ref:`ug_matter_gs_tools_matter_west_commands_append` within the :ref:`ug_matter_gs_tools_matter_west_commands` page.
  * New arguments to the :ref:`ug_matter_gs_tools_matter_west_commands_zap_tool_gui` to provide a custom cache directory and add new clusters to Matter Data Model.
  * :ref:`ug_matter_debug_snippet`.
  * Storing Matter key materials in the :ref:`matter_platforms_security_kmu`.
  * A new section :ref:`ug_matter_device_low_power_calibration_period` in the :ref:`ug_matter_device_low_power_configuration` page.
  * A new section :ref:`ug_matter_gs_tools_opp` in the :ref:`ug_matter_gs_tools` page.
  * A new overview page for :ref:`ug_nrfconnect_manufacturer_cluster_editor_tool`.

* Updated:

  * By disabling the :ref:`mpsl` before performing factory reset to speed up the process.
  * The :ref:`ug_matter_device_low_power_configuration` page to mention the `nWP049 - Matter over Thread: Power consumption and battery life`_ and `Online Power Profiler for Matter over Thread`_ as useful resources in optimizing the power consumption of a Matter device.

Matter fork
+++++++++++

> * Added a new ``kFactoryReset`` event that is posted during factory reset.

Reviewer 20: ```suggestion
The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, the ``5fd234d4f14e1225533eaea85854f160bbd0fd55`` commit from ``v1.4-branch``.

The following list summarizes the most important changes inherited from the upstream Matter:

* Added:
  * Enhanced Setup Flow, that allows the standard Matter commissioning process to enable display and acknowledgement of device makers’ legal terms and conditions prior to device setup.
  * Onboarding Payload in NFC Tags.
  * Large messages over TCP.
  * New ``kFactoryReset`` event that is posted during factory reset.
    The application can register a handler and perform additional cleanup.
```

Author: Added

  The application can register a handler and perform additional cleanup.

> * Released the first preview version of the Matter Manufacturer Cluster Editor Tool.

Reviewer 20: Theis should be moved to the `Matter` section.

Author: Moved the entry as per the suggestion.

  The tool allows you to create and edit Matter Manufacturer Cluster files or create an extension to the existing one.
  The tool is available in release artifacts.

Thread
------

* Added:

  * Support for storing the Thread key materials in the :ref:`ug_ot_thread_security_kmu`.
  * The :ref:`ug_ot_thread_security` user guide describing the security features of the |NCS| OpenThread implementation.

Zigbee
------

* Removed all Zigbee resources.
  They are now available as separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories.

Wi-Fi®
------

* The :ref:`ug_wifi_regulatory_certification` documentation is now moved under :ref:`ug_wifi` protocol page.

> 

Reviewer 21: Please add under Wi-Fi samples:
* :ref:`wifi_wfa_qt_app_sample` sample:

  * Added a new section :ref:`wifi_qt_configuration_settings`.

nRF5340 Audio
-------------

* Updated:

  * The documentation for :ref:`nrf53_audio_app_building` with cross-links and additional information.
  * The :file:`buildprog.py` script is an app-specific script for building and programming multiple kits and cores with various audio application configurations.
    The script will be deprecated in a future release.
    The audio applications will gradually shift only to using standard tools for building and programming development kits.
  * The :ref:`nRF5340 Audio application\'s <nrf53_audio_app>` :ref:`script for building and programming <nrf53_audio_app_building_script>` now builds into a directory for each transport, device type, core, and version combination.
  * The build system to use overlay files for each of the four applications instead of using :file:`Kconfig.default`.
  * The :file:`buildprog.py` script to demand argument ``--transport`` to set either ``unicast`` or ``broadcast``.

* Fixed:

  * The static random address for broadcast source and unicast server.
  * The time sync issue that occasionally caused a 1 ms difference between Left/Right headset.

nRF Desktop
-----------

* Added:

>   * System Power Management for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target on the application and radio cores.

Reviewer 15: ```suggestion
  * System Power Management for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target on the application and radio cores.
```
```suggestion
  * System power management for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target on the application and radio cores.
```

    This allows to propagate the event to further listeners of the same or lower priority.
    This prevents an issue where :ref:`nrf_desktop_ble_latency` is not informed about the connection parameter update (it might cause missing connection latency updates).
  * The Low Latency Packet Mode (LLPM) dependency in the :ref:`nrf_desktop_ble_conn_params`.
    The module relies on the :kconfig:option:`CONFIG_CAF_BLE_USE_LLPM` Kconfig option.
    This allows using the module also when the Bluetooth LE controller is not part of the main application.
  * Enabled the :ref:`CONFIG_DESKTOP_CONFIG_CHANNEL_OUT_REPORT <config_desktop_app_options>` Kconfig option for the nRF54H20 DK.
    The option mitigates HID report rate drops during DFU image transfer through the nRF Desktop dongle.
  * Explicitly enabled the :kconfig:option:`CONFIG_BT_CTLR_ASSERT_HANDLER` Kconfig option in IPC radio image configurations of the nRF54H20 DK.
    This is done to use an assertion handler defined by the IPC radio image.
  * Disabled the UDC DWC2 DMA support (:kconfig:option:`CONFIG_UDC_DWC2_DMA`) on nRF54H20 DK.
    The DMA support is experimental, disabling the feature improves USB HID stability.

>     Since nRF Desktop uses only small HID reports (report size is smaller than 64 bytes), the DMA does not improve performance.

Reviewer 4: Consider informing about rationale and making it a bit more precise:
```
  * The nRF Desktop device names to remove the "52" infix, because the nRF Desktop application supports also other SoC Series.
    This change breaks backwards compatibility.
    Peripherals using firmware from NCS 3.0.0 (or newer) won't pair with dongles using firmware from an older NCS release and the other way around.
    HID configurator script was also aligned to new naming scheme.
```

>     Since nRF Desktop uses only small HID reports (report size is smaller than 64 bytes), the DMA does not improve performance.

Reviewer 5: Okay, let's stick to @MarekPieta's version.

>     Since nRF Desktop uses only small HID reports (report size is smaller than 64 bytes), the DMA does not improve performance.

Author: Added.

Networking samples
------------------

Added support for nRF7002 EB II with ``nrf54l15dk/nrf54l15/cpuapp`` board target in the following samples:

* :ref:`aws_iot`
* :ref:`download_sample`
* :ref:`net_coap_client_sample`
* :ref:`https_client`
* :ref:`mqtt_sample`
* :ref:`udp_sample`

* Updated:

  * The :kconfig:option:`CONFIG_HEAP_MEM_POOL_SIZE` Kconfig option value to ``1280`` for all networking samples that had it set to a lower value.
    This is a requirement from Zephyr and removes a build warning.
  * The following samples to use the :ref:`lib_downloader` library instead of the :ref:`lib_download_client` library:

    * :ref:`aws_iot`
    * :ref:`azure_iot_hub`
    * :ref:`download_sample`

NFC samples
-----------

* Added support for the ``nrf54l15dk/nrf54l05/cpuapp`` and ``nrf54l15dk/nrf54l10/cpuapp`` board targets in the following samples:

    * :ref:`record_text`
    * :ref:`record_launch_app`
    * :ref:`nfc_shell`
    * :ref:`nrf-nfc-system-off-sample`
    * :ref:`nfc_tnep_tag`
    * :ref:`writable_ndef_msg`

nRF5340 samples
---------------

* Removed the nRF5340: Multiprotocol RPMsg sample.
  Use the :ref:`ipc_radio` application instead.

> 

Reviewer 18: Please add:

* :ref:`smp_svr_ext_xip` sample
  * Added:
 nRF5340 non-secure (TF-M) target

Author: Added

Peripheral samples
------------------

* :ref:`radio_test` sample:

  * Added:

    * Loading of radio trims and a fix of a hardware errata for the nRF54H20 SoC to improve the RF performance.
    * Support for the :ref:`nRF2220 front-end module <ug_radio_fem_nrf2220ek>`.
    * Workaround for the hardware errata HMPAN-216 for the nRF54H20 SoC.

PMIC samples
------------

* Added:

  * The :ref:`npm2100_one_button` sample that demonstrates how to support wake-up, shutdown, and user interactions through a single button connected to the nPM2100 PMIC.
  * The :ref:`npm2100_fuel_gauge` sample that demonstrates how to calculate the battery state of charge of primary cell batteries using the :ref:`nrfxlib:nrf_fuel_gauge`.

* :ref:`npm1300_fuel_gauge` sample:

  * Updated to accommodate API changes in nRF Fuel Gauge library v1.0.0.

SDFW samples
------------

* Removed the SDFW: Service Framework Client sample as all services demonstrated by the sample have been removed.

SUIT samples
------------

* Added the :ref:`nrf54h_suit_ab_sample` sample that demonstrates how to perform A/B updates using SUIT manifests.

>  * :ref:`nrf54h_suit_sample` sample:

Author: Fixed

   * Updated:

     * The memory maps to cover the entire available MRAM memory.
     * The memory maps to place recovery firmware on lower addresses than the main firmware.
     * By enabling secure entropy source in all main Bluetooth-enabled sample variants (except recovery firmware).
     * By extending the manifests to process the ``suit-payload-fetch`` sequence of the Nordic top update candidate.

>      * By extending the manifests with build-time checks for consistency between MPI and envelope signing configuration.

Author: Added

Other samples
-------------

* Added :ref:`app_jwt_sample` sample that demonstrates how the application core can generate a signed JWT.

* :ref:`coremark_sample` sample:

  * Added:

    * Support for the nRF54L05 and nRF54L10 SoCs (emulated on nRF54L15 DK).
    * FLPR core support for the :ref:`zephyr:nrf54l15dk_nrf54l15` and :ref:`zephyr:nrf54h20dk_nrf54h20` board targets.

  * Removed the following compiler options that were set in the :kconfig:option:`CONFIG_COMPILER_OPT` Kconfig option:

    * ``-fno-pie``
    * ``-fno-pic``
    * ``-ffunction-sections``
    * ``-fdata-sections``

    These options are enabled by default in Zephyr and do not need to be set with the dedicated Kconfig option.

* :ref:`caf_sensor_manager_sample` sample:

  * Added low power configuration for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target.

Drivers
=======

This section provides detailed lists of changes by :ref:`driver <drivers>`.

* Added a :ref:`flash_ipuc` that allows to manage SUIT IPUC memory through the Zephyr flash API.

Wi-Fi drivers
-------------

* Added:

  * Advanced debug shell for reading and writing registers and memory of the nRF70 Series chip.

>     The debug shell can be enabled using the :kconfig:option:`CONFIG_WIFI_NRF70_DEBUG_SHELL` Kconfig option.

Reviewer 15: ```suggestion
    The debug shell can be enabled using the :kconfig:option:`CONFIG_NRF70_DEBUG_SHELL` Kconfig option.
```

Reviewer 7: @krish2718 Could you please confirm?

    * The method of supplying the Fast Pair Model ID and Anti-Spoofing Private Key to generate the Fast Pair provisioning data HEX file.
      The ``FP_MODEL_ID`` and ``FP_ANTI_SPOOFING_KEY`` CMake variables are replaced by the corresponding ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig options.
    * The automatically generated ``bt_fast_pair`` partition definition (in the :file:`subsys/partition_manager/pm.yml.bt_fast_pair` file) to work correctly when building with TF-M.
    * The behavior of the :c:member:`bt_fast_pair_fmdn_info_cb.provisioning_state_changed` callback.
      The callback no longer reports the initial provisioning state after the Fast Pair subsystem is enabled with the :c:func:`bt_fast_pair_enable` function call.
      See the :ref:`migration guide <migration_3.0_recommended>` for mandatory changes and the :ref:`ug_bt_fast_pair_gatt_service_fmdn_info_callbacks_provisioning_state` section in the Fast Pair integration guide for the description on how to track the FMDN provisioning state with the new approach.

>   * Removed the sysbuild control over the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is defined in the main (default) image.
>      Sysbuild no longer sets the value of this Kconfig option.

Reviewer 7: ```suggestion
  * Removed the sysbuild control over the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is defined in the main (default) image.
    Sysbuild no longer sets the value of this Kconfig option.
```

>   * Updated:
> 
>     * The :c:func:`bt_fast_pair_info_cb_register` API to allow registration of multiple callbacks.
>     * The Fast Pair sysbuild Kconfig options.
>       The ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option is replaced with the ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY``.
>     * The method of supplying the Fast Pair Model ID and Anti-Spoofing Private Key to generate the Fast Pair provisioning data HEX file.
>       The ``FP_MODEL_ID`` and ``FP_ANTI_SPOOFING_KEY`` CMake variables are replaced by the corresponding ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig options.
>     * The automatically generated ``bt_fast_pair`` partition definition (in the :file:`subsys/partition_manager/pm.yml.bt_fast_pair` file) to work correctly when building with TF-M.
>     * The behavior of the :c:member:`bt_fast_pair_fmdn_info_cb.provisioning_state_changed` callback.
>       The callback no longer reports the initial provisioning state after the Fast Pair subsystem is enabled with the :c:func:`bt_fast_pair_enable` function call.
>       See the :ref:`migration guide <migration_3.0_recommended>` for mandatory changes and the :ref:`ug_bt_fast_pair_gatt_service_fmdn_info_callbacks_provisioning_state` section in the Fast Pair integration guide for the description on how to track the FMDN provisioning state with the new approach.
> 
>   * Removed the sysbuild control over the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is defined in the main (default) image.
>      Sysbuild no longer sets the value of this Kconfig option.

Reviewer 15: ```suggestion
* Updated:

  * The :c:func:`bt_fast_pair_info_cb_register` API to allow registration of multiple callbacks.
  * The Fast Pair sysbuild Kconfig options.
    The ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option is replaced with the ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY``.
  * The method of supplying the Fast Pair Model ID and Anti-Spoofing Private Key to generate the Fast Pair provisioning data HEX file.
    The ``FP_MODEL_ID`` and ``FP_ANTI_SPOOFING_KEY`` CMake variables are replaced by the corresponding ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig options.
  * The automatically generated ``bt_fast_pair`` partition definition (in the :file:`subsys/partition_manager/pm.yml.bt_fast_pair` file) to work correctly when building with TF-M.
  * The behavior of the :c:member:`bt_fast_pair_fmdn_info_cb.provisioning_state_changed` callback.
    The callback no longer reports the initial provisioning state after the Fast Pair subsystem is enabled with the :c:func:`bt_fast_pair_enable` function call.
    See the :ref:`migration guide <migration_3.0_recommended>` for mandatory changes and the :ref:`ug_bt_fast_pair_gatt_service_fmdn_info_callbacks_provisioning_state` section in the Fast Pair integration guide for the description on how to track the FMDN provisioning state with the new approach.

* Removed the sysbuild control over the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is defined in the main (default) image.
  Sysbuild no longer sets the value of this Kconfig option.
```

  * Fixed a bug where various subsystems would be erroneously initialized during a failed initialization of the library.

* :ref:`lib_location` library:

  * Removed support for HERE location services.

* :ref:`lib_at_host` library:

  * Fixed a bug where AT responses would erroneously be written to the logging UART instead of being written to the chosen ``ncs,at-host-uart`` UART device when the :kconfig:option:`CONFIG_LOG_BACKEND_UART` Kconfig option was set.

* :ref:`modem_info_readme` library:

  * Added:

    * The :c:enum:`modem_info_data_type` type for representing LTE link information data types.
    * The :c:func:`modem_info_data_type_get` function for requesting the data type of the current modem information type.

  * Deprecated the :c:func:`modem_info_type_get` function in favor of the :c:func:`modem_info_data_type_get` function.

* :ref:`lib_modem_slm` library:

  * Updated:

    * By renaming the ``CONFIG_MODEM_SLM_WAKEUP_PIN`` and ``CONFIG_MODEM_SLM_WAKEUP_TIME`` Kconfig options to :kconfig:option:`CONFIG_MODEM_SLM_POWER_PIN` and :kconfig:option:`CONFIG_MODEM_SLM_POWER_PIN_TIME`, respectively.

>     * By renaming the :c:func:`modem_slm_wake_up` function to :c:func:`modem_slm_power_pin_toggle`.

Reviewer 15: ```suggestion
    * By renaming the ``modem_slm_wake_up()`` function to :c:func:`modem_slm_power_pin_toggle`.
```

  Only supported for nRF91 Series devices.

* Added a new feature to automatically capture and upload modem traces to Memfault with coredumps upon a crash.
  To enable this feature, set the :kconfig:option:`CONFIG_MEMFAULT_NCS_POST_MODEM_TRACE_ON_COREDUMP` Kconfig option to ``y``.
  Only supported for nRF91 Series devices.

sdk-nrfxlib
-----------

See the changelog for each library in the :doc:`nrfxlib documentation <nrfxlib:README>` for additional information.

* Removed Zboss documentation.

MCUboot
=======

The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``a2bc982b3379d51fefda3e17a6a067342dce1a8b``, with some |NCS| specific additions.

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

* Fixed an issue where an unusable secondary slot was cleared three times instead of once during cleanup.

> 

Reviewer 18: Added keys revocation scheme support for nRF54L SoCs, see ``CONFIG_BOOT_KEYS_REVOCATION`` MCUboot Kconfig option.
Improved time performance of firmware update by usage of best RRAM write operation buffering on nRF54L SoCs.
Introduced improved swap algorithm: swap-using-offset, see ``CONFIG_BOOT_SWAP_USING_OFFSET`` MCUboot Kconfig option.
Image compression support has been brought to production quality

Author: Added

  * An issue where Bluetooth applications built with the ``nordic-bt-rpc`` snippet (in the :ref:`ble_rpc` configuration) did not work on the nRF54H20 devices due to incorrect memory mapping.

* Removed the ``HCI_LE_Read_Local_P-256_Public_Key`` and ``HCI_LE_Generate_DHKey`` commands emulation from HCI driver.

Bluetooth Mesh
--------------

* Added the key importer functionality (:kconfig:option:`CONFIG_BT_MESH_KEY_IMPORTER`).
* Updated the default value for :kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE` if :kconfig:option:`CONFIG_BT_MESH_NLC_PERF_CONF` is selected.
  The :kconfig:option:`CONFIG_BT_MESH_NLC_PERF_CONF` Kconfig option increases the number of keys used by Mesh and the value of :kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE` Kconfig option needs to be increased accordingly.
* Deprecation procedure has started for :kconfig:option:`CONFIG_BT_MESH_USES_TINYCYPT` and it is not recommended for future designs.
  For platforms that support TF-M, the :kconfig:option:`CONFIG_BT_MESH_USES_TFM_PSA` Kconfig option is used by default.
  For platforms that do not support TF-M, the :kconfig:option:`CONFIG_BT_MESH_USES_MBEDTLS_PSA` Kconfig option is used by default.

> 

Reviewer 22: Following needs to be added:
```suggestion
Removed experimental flags for TF-M PSA and mbed TLS PSA.
```

    * :ref:`ble_mesh_dfu_distributor`

  * Support for nRF54L05 in the following samples:

    * :ref:`bluetooth_mesh_sensor_client`
    * :ref:`bluetooth_mesh_sensor_server`
    * :ref:`bluetooth_ble_peripheral_lbs_coex`
    * :ref:`bt_mesh_chat`
    * :ref:`bluetooth_mesh_light_switch`
    * :ref:`bluetooth_mesh_silvair_enocean`
    * :ref:`bluetooth_mesh_light_dim`
    * :ref:`bluetooth_mesh_light`
    * :ref:`bluetooth_mesh_light_lc`

* Updated:

  * The board configuration files for targets nRF54L15, nRF54L10 and nRF54L05 by increasing the values of :kconfig:option:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` and :kconfig:option:`CONFIG_BT_RX_STACK_SIZE` Kconfig options to prevent stack overflows when :kconfig:option:`CONFIG_BT_MESH_USES_MBEDTLS_PSA` is selected.
  * The following samples to include the value of the :kconfig:option:`CONFIG_BT_COMPANY_ID` option in the Firmware ID:

    * :ref:`ble_mesh_dfu_distributor`
    * :ref:`ble_mesh_dfu_target`

  * The friend feature is disabled when the :ref:`bluetooth_mesh_light_lc` sample is compiled for the :ref:`zephyr:nrf52dk_nrf52832` board target to increase the amount of RAM available for the application.

> * :ref:`bluetooth_mesh_light_lc` sample:

Reviewer 22: We should remove this last item, it is the same with the previous one.
```suggestion
```

      This is part of the ongoing work of archiving `nRF Command Line Tools`_ and replacing them with nRF Util.
    * The Running applications with Trusted Firmware-M page by renaming it to :ref:`ug_tfm` and moving it under :ref:`ug_tfm_index`.
    * The :ref:`app_boards_spe_nspe` documentation page from the :ref:`ug_app_dev` section has been moved under :ref:`ug_tfm_index`.

Protocols
=========

|no_changes_yet_note|

Amazon Sidewalk
---------------

|no_changes_yet_note|

Bluetooth® LE
-------------

* Updated the Bluetooth LE SoftDevice Controller driver to make the :c:func:`hci_vs_sdc_llpm_mode_set` function return an error if Low Latency Packet Mode (LLPM) is not supported or not enabled in the Bluetooth LE Controller driver configuration (:kconfig:option:`CONFIG_BT_CTLR_SDC_LLPM`).

* Fixed:

  * An issue where a flash operation executed on the system workqueue might result in ``-ETIMEDOUT``, if there is an active Bluetooth LE connection.
  * An issue where Bluetooth applications built with the ``nordic-bt-rpc`` snippet (in the :ref:`ble_rpc` configuration) did not work on the nRF54H20 devices due to incorrect memory mapping.

Bluetooth Mesh
--------------

> * Added the key importer functionality (:kconfig:option:`CONFIG_BT_MESH_KEY_IMPORTER`).

Reviewer 22: ```suggestion
* Deprecation procedure has started for (:kconfig:option:`CONFIG_BT_MESH_USES_TINYCYPT`) and it is not recommended for future desings. For platforms that support TF-M (:kconfig:option:`CONFIG_BT_MESH_USES_TFM_PSA`) is used by default, for platforms that do not support TF-M (:kconfig:config:`CONFIG_BT_MESH_USES_MBEDTLS_PSA`) is used by default.
* Added the key importer functionality (:kconfig:option:`CONFIG_BT_MESH_KEY_IMPORTER`).
* Updated the default (:kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE`) if (:kconfig:option:`BT_MESH_NLC_PERF_CONF`) is selected. (:kconfig:option:`BT_MESH_NLC_PERF_CONF`) increases the number of keys used by mesh and (:kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE`) needs to be increased accordingly.
```

Author: Added with minor modifications. Please check.

  * Added reconnection to bonded devices based on their address.

* :ref:`power_profiling` sample:

  * Added the :kconfig:option:`CONFIG_BT_POWER_PROFILING_NFC_DISABLED` Kconfig option to reduce the power consumption by disabling the NFC.

Bluetooth Mesh samples
----------------------

* Added:

  * Support for nRF54L10 in the following samples:

    * :ref:`bluetooth_mesh_sensor_client`
    * :ref:`bluetooth_mesh_sensor_server`
    * :ref:`bluetooth_ble_peripheral_lbs_coex`
    * :ref:`bt_mesh_chat`
    * :ref:`bluetooth_mesh_light_switch`
    * :ref:`bluetooth_mesh_silvair_enocean`
    * :ref:`bluetooth_mesh_light_dim`
    * :ref:`bluetooth_mesh_light`
    * :ref:`ble_mesh_dfu_target`
    * :ref:`bluetooth_mesh_light_lc`
    * :ref:`ble_mesh_dfu_distributor`

  * Support for nRF54L05 in the following samples:

    * :ref:`bluetooth_mesh_sensor_client`
    * :ref:`bluetooth_mesh_sensor_server`
    * :ref:`bluetooth_ble_peripheral_lbs_coex`
    * :ref:`bt_mesh_chat`
    * :ref:`bluetooth_mesh_light_switch`
    * :ref:`bluetooth_mesh_silvair_enocean`
    * :ref:`bluetooth_mesh_light_dim`
    * :ref:`bluetooth_mesh_light`
    * :ref:`bluetooth_mesh_light_lc`

> * Updated the following samples to include the value of the :kconfig:option:`CONFIG_BT_COMPANY_ID` option in the Firmware ID:

Reviewer 22: ```suggestion
* Updated:

  * The board configuration files for targets nRF54L15, nRF54L10 and nRF54L05 by increasing (:kconfig:config:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE`) and (:kconfig:config:`CONFIG_BT_RX_STACK_SIZE`) to prevent stack overflows when (:kconfig:config:`CONFIG_BT_MESH_USES_MBEDTLS_PSA`) is selcted.

  * The following samples to include the value of the :kconfig:option:`CONFIG_BT_COMPANY_ID` option in the Firmware ID:

    * :ref:`ble_mesh_dfu_distributor`
    * :ref:`ble_mesh_dfu_target`

  * Disabling the friend feature when the following sample is compiled for the :ref:`zephyr:nrf52dk_nrf52832` board target to increase the amount of RAM available for the application.

    * :ref:`bluetooth_mesh_light_lc`
```

* :kconfig:option:`CONFIG_SETTINGS_ZMS_LL_CACHE_SIZE` - The size of the linked list cache (each entry occupies 8 Bytes of RAM).
* :kconfig:option:`CONFIG_SETTINGS_ZMS_NO_LL_DELETE` - Disables deleting the linked list nodes when deleting a Settings Key.
  Use this option only when the application is always using the same Settings Keys.
  When the application uses random Keys, enabling this option could lead to incrementing the linked list nodes without corresponding Keys and cause excessive delays to loading of the Keys.
  Use this option only to accelerate the delete operation for a fixed set of Settings elements.
* :kconfig:option:`CONFIG_SETTINGS_ZMS_LOAD_SUBTREE_PATH` - Loads first the subtree path passed in the argument, then continue to load all the Keys in the same subtree if the handler returns a zero value.
* Using :c:func:`settings_load_one` function to retrieve the value of a known Key is recommended with ZMS as it improves the read performance a lot.
* To get the value length of a Settings entry, it is recommended to use the :c:func:`settings_get_val_len` function.

zcbor
=====

|no_changes_yet_note|

Trusted Firmware-M
==================

|no_changes_yet_note|

cJSON

> =====

Author: @anangl, could you check if there is anything to be added here?

Author: Removed the section after discussing with @carlescufi .

Build and configuration system
==============================

* Removed support for the deprecated multi-image builds (parent-child images) functionality.
  All |NCS| projects must now use :ref:`sysbuild`.
  See :ref:`child_parent_to_sysbuild_migration` for an overview of differences with parent-child image and how to migrate.
* Updated:

  * The default runner for the ``west flash`` command to `nRF Util`_ instead of ``nrfjprog`` that is part of the archived `nRF Command Line Tools`_.
    For more information, see the :ref:`build system section in the v3.0.0 migration guide <migration_3.0_recommended>` and the :ref:`programming_selecting_runner` section on the programming page.
  * Erasing the external memory when programming a new firmware image with the ``west flash`` series now always correctly honors the ``--erase`` flag (and its absence) both when using the ``nrfjprog`` and ``nrfutil`` backends.
    Prior to this release, the ``nrfjprog`` backend would always erase only the sectors of the external flash used by the new firmware, and the ``nrfutil`` backend would always erase the whole external flash.

>   * The ``CONFIG_NRF53_MULTI_IMAGE_UPDATE`` Kconfig option no longer depends on external flash (NCSIDB-1232).

Author: Discussed offline.

  * The static partition manager file for network core images can now be set (NCSIDB-1442).
  * QSPI XIP support has been extended to include building with TF-M on the nRF5340 device.
    An instance of TF-M can be now part of the internal NVM application image.

Bootloader and DFU
==================

* Added experimental support |NSIB| for the nRF54L15 SoC.

>   On nRF54L SoCs, NSIB uses KMU for authentication key storage, it supports keys revocation scheme and it can be protected using immutable-boot region SoC's hardware feature.

Author: @annwoj, please check the changes made.

* Renamed the SDP GPIO application to :ref:`hpf_gpio_example`.

Machine learning
----------------

* Updated:

  * The application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system for the :ref:`zephyr:nrf54h20dk_nrf54h20` board.
  * The Edge Impulse URI configuration to use the new model location.

IPC radio firmware
------------------

* Updated:

  * The application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system in all devices that contain MRAM, such as the nRF54H Series devices.
  * The documentation of applications and samples that use the IPC radio firmware as a :ref:`companion component <companion_components>` to mention its usage when built with :ref:`configuration_system_overview_sysbuild`.

Matter bridge
-------------

* Updated by enabling Link Time Optimization (LTO) by default for the ``release`` configuration.
* Removed support for the nRF54H20 devices.

nRF5340 Audio
-------------

* Updated:

  * The documentation for :ref:`nrf53_audio_app_building` with cross-links and additional information.

>   * The :file:`buildprog.py` script is an app-specific script for building and programming multiple kits and cores with various audio application configurations.

Reviewer 12: Looks good

    The feature is enabled by default if Zephyr power management (:kconfig:option:`CONFIG_PM`) is enabled.
    It prevents entering power states that introduce wakeup latency and ensures high performance.
  * Static Partition Manager memory maps for single-image configurations (without bootloader and separate radio/network core image).
    In the |NCS|, the Partition Manager is enabled by default for single-image sysbuild builds.
    The static memory map ensures control over settings partition placement and size.
    The introduced static memory maps might not be consistent with the ``storage_partition`` defined by the board-level DTS configuration.
  * Support for GATT long (reliable) writes (:kconfig:option:`CONFIG_BT_ATT_PREPARE_COUNT`) to Fast Pair and Works With ChromeBook (WWCB) configurations.
    This allows performing :ref:`fwupd <nrf_desktop_fwupd>` DFU image upload over Bluetooth LE with GATT clients that do not perform MTU exchange (for example, ChromeOS using the Floss Bluetooth stack).
  * The ``dongle`` and ``release_dongle`` application configurations for the nRF54H20 Development Kit (``nrf54h20dk/nrf54h20/cpuapp``).
    The configurations act as a HID dongle.

> * Updated:

Reviewer 4: https://github.com/nrfconnect/sdk-nrf/pull/21886 was already merged

> * Updated:

Author: Added

  * Enabled the :ref:`CONFIG_DESKTOP_CONFIG_CHANNEL_OUT_REPORT <config_desktop_app_options>` Kconfig option for the nRF54H20 DK.
    The option mitigates HID report rate drops during DFU image transfer through the nRF Desktop dongle.
  * Explicitly enabled the :kconfig:option:`CONFIG_BT_CTLR_ASSERT_HANDLER` Kconfig option in IPC radio image configurations of the nRF54H20 DK.
    This is done to use an assertion handler defined by the IPC radio image.
  * Disabled the UDC DWC2 DMA support (:kconfig:option:`CONFIG_UDC_DWC2_DMA`) on the nRF54H20 DK.
    The DMA support is experimental, and disabling the feature improves USB HID stability.
    Since nRF Desktop uses only small HID reports (report size is smaller than 64 bytes), the DMA does not improve performance.
  * The nRF Desktop device names to remove the``52`` infix, because the nRF Desktop application supports other SoC Series also.
    This change breaks backwards compatibility.
    Peripherals using firmware from |NCS| 3.0.0 (or newer) will not pair with dongles using firmware from an older |NCS| release and the other way around.
    The HID configurator script has also been aligned to the new naming scheme.

* Removed:

>   * An imply from the nRF Desktop Bluetooth connectivity Kconfig option (:ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>`).

Reviewer 4: Please remove the:

```
  * An imply from the nRF Desktop Bluetooth connectivity Kconfig option (:ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>`).
    The imply enabled a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) if MPSL was used for synchronization between the flash memory driver and the radio (:kconfig:option:`CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL`).
    The workaround for the MPSL flash synchronization issue (NCSDK-29354 in the :ref:`known_issues` page) is no longer needed, as the issue is now fixed.
```
The change was enabled back here: https://github.com/nrfconnect/sdk-nrf/pull/21868

Author: Removed this entry.

  * Support for nRF54L05 in the following samples:

    * :ref:`bluetooth_mesh_sensor_client`
    * :ref:`bluetooth_mesh_sensor_server`
    * :ref:`bluetooth_ble_peripheral_lbs_coex`
    * :ref:`bt_mesh_chat`
    * :ref:`bluetooth_mesh_light_switch`
    * :ref:`bluetooth_mesh_silvair_enocean`
    * :ref:`bluetooth_mesh_light_dim`
    * :ref:`bluetooth_mesh_light`
    * :ref:`bluetooth_mesh_light_lc`

* Updated:

  * The board configuration files for targets nRF54L15, nRF54L10, and nRF54L05 by increasing the values of :kconfig:option:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` and :kconfig:option:`CONFIG_BT_RX_STACK_SIZE` Kconfig options to prevent stack overflows when :kconfig:option:`CONFIG_BT_MESH_USES_MBEDTLS_PSA` is selected.
  * The following samples to include the value of the :kconfig:option:`CONFIG_BT_COMPANY_ID` option in the Firmware ID:

    * :ref:`ble_mesh_dfu_distributor`
    * :ref:`ble_mesh_dfu_target`

  * The friend feature is disabled when the :ref:`bluetooth_mesh_light_lc` sample is compiled for the :ref:`zephyr:nrf52dk_nrf52832` board target to increase the amount of RAM available for the application.

Bluetooth Fast Pair samples
---------------------------

> * Added support for the ``nrf54l15dk/nrf54l05/cpuapp`` and ``nrf54l15dk/nrf54l10/cpuapp`` board targets in all Fast Pair samples.

Reviewer 6: ```suggestion
* Added experimental support for the ``nrf54l15dk/nrf54l05/cpuapp`` and ``nrf54l15dk/nrf54l10/cpuapp`` board targets in all Fast Pair samples.
```

    This change results from the Bluetooth subsystem transition to the PSA cryptographic standard.
    The Bluetooth stack can now use the PSA crypto API in the non-secure domain as all necessary TF-M partitions are configured properly.
  * The configuration of all Fast Pair samples by increasing the following thread stack sizes to prevent stack overflows:

    * The system workqueue thread stack (:kconfig:option:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE`).
    * The Bluetooth receiving thread stack (:kconfig:option:`CONFIG_BT_RX_STACK_SIZE`).

    This change results from the Bluetooth subsystem transition to the PSA cryptographic API.
  * The sysbuild configurations in samples to align them with the sysbuild Kconfig changes for controlling the Fast Pair provisioning process.

* Removed using a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) from configurations.
  The MPSL flash synchronization issue (NCSDK-29354 in the :ref:`known_issues`) is fixed.
  The workaround is no longer needed.

* :ref:`fast_pair_locator_tag` sample:

  * Added:

>     * Support for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target.

Author: Added

       * To declare an IPUC as DFU cache area.
       * To trigger Nordic firmware updates from an IPUC-based DFU cache area.

     * Support for IPUC in DFU protocols that makes possible the following:

       * To write into IPUC using SMP image command group.
       * To write into IPUC using SUIT SMP cache raw upload commands.
       * To write into IPUC using SUIT :ref:`dfu_target<lib_dfu_target_suit_style_update>` library.

     * Possibilities for the following:

       * To copy binaries into radio local RAM memory from SUIT radio manifests.
       * To specify the minimal Nordic top manifest version in the :file:`VERSION` file.
       * To block independent updates of Nordic manifests using the :kconfig:option:`CONFIG_SUIT_NORDIC_TOP_INDEPENDENT_UPDATE_FORBIDDEN` Kconfig option.

   * Updated by moving the MPI configuration from local Kconfig options to sysbuild.

Modem libraries
---------------

* Deprecated the :ref:`at_params_readme` library.

* :ref:`pdn_readme` library:

  * Deprecated the :c:func:`pdn_dynamic_params_get` function.
    Use the new function :c:func:`pdn_dynamic_info_get` instead.

* :ref:`lte_lc_readme` library:

>   * Fixed handling of ``%NCELLMEAS`` notification with status 2 (measurement interrupted) and no cells.
>   * Added sending of ``LTE_LC_EVT_NEIGHBOR_CELL_MEAS`` event with ``current_cell`` set to ``LTE_LC_CELL_EUTRAN_ID_INVALID`` in case an error occurs while parsing the ``%NCELLMEAS`` notification.

Reviewer 17: ```suggestion
  * Added sending of ``LTE_LC_EVT_NEIGHBOR_CELL_MEAS`` event with ``current_cell`` set to ``LTE_LC_CELL_EUTRAN_ID_INVALID`` in case an error occurs while parsing the ``%NCELLMEAS`` notification.
  * Fixed handling of ``%NCELLMEAS`` notification with status 2 (measurement interrupted) and no cells.
```

    * By renaming the ``modem_slm_wake_up`` function to :c:func:`modem_slm_power_pin_toggle`.

Multiprotocol Service Layer libraries
-------------------------------------

* Added:

  * Integration with the nrf2 clock control driver for the nRF54H20 SoC.
  * Integration with Zephyr's system power management for the nRF54H20 SoC.
  * Global domain HSFLL120 320MHz frequency request if MPSL is enabled.
    The high frequency in global domain is required to ensure that fetching instructions from L2-cache and MRAM is as fast as possible.
    It is needed for the radio protocols to operate correctly.
  * MRAM always-on request for scheduled radio events.
    It is needed to avoid MRAM wake-up latency for radio protocols.

Libraries for networking
------------------------

* Added:

  * The :ref:`lib_downloader` library.
  * A backend for the :ref:`TLS Credentials Subsystem <zephyr:sockets_tls_credentials_subsys>` that stores the credentials in the modem, see :kconfig:option:`CONFIG_TLS_CREDENTIALS_BACKEND_NRF_MODEM`.

* Deprecated the :ref:`lib_download_client` library.

>   See the :ref:`migration guide <migration_3.0_recommended>` for recommended changes.

Reviewer 17: I believe this entry should be after the Updated entry.

  * :ref:`lib_nrf_cloud`
  * :ref:`lib_aws_fota`
  * :ref:`lib_azure_fota`
  * :ref:`lib_fota_download`

* :ref:`lib_nrf_cloud_pgps` library:

  * Fixed the warning due to missing ``https`` download protocol.

* :ref:`lib_downloader` library:

  * Updated to support Proxy-URI option and an authentication callback after connecting.

* :ref:`lib_fota_download` library:

  * Added error codes related to unsupported protocol, DFU failures, and invalid configuration.
  * Updated to use the :ref:`lib_downloader` library for CoAP downloads.

* :ref:`lib_nrf_cloud` library:

  * Added the :kconfig:option:`CONFIG_NRF_CLOUD` Kconfig option to prevent unintended inclusion of nRF Cloud Kconfig variables in non-nRF Cloud projects.

  * Updated:

    * To use the :ref:`lib_downloader` library for CoAP downloads.
    * To use the :ref:`lib_app_jwt` library to generate JWT tokens.

Other libraries
---------------

* Added new library :ref:`lib_app_jwt` library.

* :ref:`mod_dm` library:

  * Updated the default timeslot duration to avoid an overstay assert when the ranging failed.

> * Removed the following unused SDFW services: ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.

Reviewer 17: ```suggestion
* Removed the unused SDFW services ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.
```

    The Bluetooth stack can now use the PSA crypto API in the non-secure domain as all necessary TF-M partitions are configured properly.

  * The configurations of the following samples by increasing the main thread stack size (:kconfig:option:`CONFIG_MAIN_STACK_SIZE`) to prevent stack overflows:

    * :ref:`bluetooth_central_hids`
    * :ref:`peripheral_hids_keyboard`
    * :ref:`peripheral_hids_mouse`

    This change results from the Bluetooth subsystem transition to the PSA cryptographic API.

  * The following samples to use LE Secure Connection pairing (:kconfig:option:`CONFIG_BT_SMP_SC_PAIR_ONLY`).

    * :ref:`peripheral_gatt_dm`
    * :ref:`peripheral_mds`
    * :ref:`peripheral_cts_client`

* :ref:`direct_test_mode` sample:

  * Added:

    * Loading of radio trims and a fix of a hardware errata for the nRF54H20 SoC to improve the RF performance.
    * Support for the :ref:`nRF2220 front-end module <ug_radio_fem_nrf2220ek>`.
    * Workaround for the hardware errata HMPAN-216 for the nRF54H20 SoC.

* :ref:`central_uart` sample:

  * Added reconnection to bonded devices based on their address.

> 

Reviewer 6: Please add the following note that describes the change done in the following PR:

https://github.com/nrfconnect/sdk-nrf/pull/21844

```suggestion

* :ref:`peripheral_hids_keyboard` sample:

  * Fixed the issue with the :kconfig:option:`CONFIG_NFC_OOB_PAIRING` Kconfig option that is defined at the sample level and could not be enabled due to the unmet dependency on the :kconfig:option:`CONFIG_HAS_HW_NRF_NFCT` Kconfig option.
    The issue is resolved by enabling the ``nfct`` node in the sample devicetree configuration, which sets the :kconfig:option:`CONFIG_HAS_HW_NRF_NFCT` Kconfig option to the expected value.
```

Author: Added

Trusted Firmware-M (TF-M) samples
---------------------------------

* :ref:`tfm_psa_template` sample:

  * Added support for the following attestation token fields:

    * Profile definition
    * PSA certificate reference (optional), configured using the ``SB_CONFIG_TFM_OTP_PSA_CERTIFICATE_REFERENCE`` sysbuild Kconfig option
    * Verification service URL (optional), configured using the :kconfig:option:`CONFIG_TFM_ATTEST_VERIFICATION_SERVICE_URL` Kconfig option

* :ref:`tfm_secure_peripheral_partition` sample:

  * Updated documentation with information about how to access other TF-M partitions from the secure partition.

Thread samples
--------------

* :ref:`ot_cli_sample` sample:

  * Removed support for the nRF54H20 DK.

Zigbee samples
--------------

* Removed all Zigbee samples.
  They are now available as separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories.

Wi-Fi samples
-------------

* Added support for nRF7002-EB II with nRF54 Series devices in the following samples:

  * :ref:`wifi_station_sample`
  * :ref:`wifi_scan_sample`
  * :ref:`wifi_shell_sample`
  * :ref:`wifi_radio_test_sd`
  * :ref:`wifi_radio_test`
  * :ref:`wifi_softap_sample`

>   * :ref:`wifi_shutdown_sample`

Reviewer 8: Please remove this line.

Author: Removed

The following list summarizes the most important changes inherited from the upstream Matter:

* Added:

  * Enhanced Setup Flow, that allows the standard Matter commissioning process to enable display and acknowledgment of device makers' legal terms and conditions prior to device setup.
  * Onboarding Payload in NFC Tags.
  * Large messages over TCP.
  * New ``kFactoryReset`` event that is posted during factory reset.
    The application can register a handler and perform additional cleanup

Thread
------

* Added:

  * Support for storing the Thread key materials in the :ref:`ug_ot_thread_security_kmu`.
  * The :ref:`ug_ot_thread_security` user guide describing the security features of the |NCS| OpenThread implementation.

Zigbee
------

* Removed all Zigbee resources.
  They are now available as separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories.

Wi-Fi®
------

* Modified Throughputs for Wi-Fi Usage profiles.
* Wi-Fi cred shell has been renamed from ``wifi_cred`` to ``wifi cred``.
* The :ref:`ug_wifi_regulatory_certification` documentation is now moved under :ref:`ug_wifi` protocol page.

Applications
============

> * Removed the Asset Tracker v2 application.

Reviewer 23: ```suggestion
* Added the new :ref:`hpf_mspi_example` application.

* Removed the Asset Tracker v2 application.
```

Author: Added

    Peripherals using firmware from |NCS| 3.0.0 (or newer) will not pair with dongles using firmware from an older |NCS| release and the other way around.
    HID configurator script has also been aligned to the new naming scheme.

* Removed:

  * An imply from the nRF Desktop Bluetooth connectivity Kconfig option (:ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>`).
    The imply enabled a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) if MPSL was used for synchronization between the flash memory driver and the radio (:kconfig:option:`CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL`).
    The workaround for the MPSL flash synchronization issue (NCSDK-29354 in the :ref:`known_issues` page) is no longer needed, as the issue is now fixed.
  * Application configurations for the nRF52810 Desktop Mouse board (``nrf52810dmouse/nrf52810``).
    The board is no longer supported in the |NCS|.

Serial LTE modem
----------------

* Added a new page :ref:`slm_as_linux_modem`.
* Added an overlay file :file:`overlay-memfault.conf` to enable Memfault.
  See :ref:`mod_memfault` for more information about Memfault features in |NCS|.

> * Updated:
> 
>   * The application to use the :ref:`lib_downloader` library instead of the deprecated :ref:`lib_download_client` library.
>   * In Zephyr, the numerical values of various |NCS| specific socket options that are used with the ``#XSOCKETOPT`` command:
> 
>       * :c:macro:`TLS_DTLS_HANDSHAKE_TIMEO`- Changed from ``18`` to ``1018``.
>       * :c:macro:`SO_SILENCE_ALL` - Changed from ``30`` to ``1030``.
>       * :c:macro:`SO_IP_ECHO_REPLY` - Changed from ``31`` to ``1031``.
>       * :c:macro:`SO_IPV6_ECHO_REPLY` - Changed from ``32`` to ``1032``.
>       * :c:macro:`SO_BINDTOPDN` - Changed from ``40`` to ``1040``.
>       * :c:macro:`SO_TCP_SRV_SESSTIMEO` - Changed from ``55`` to ``1055``.
>       * :c:macro:`SO_RAI` - Changed from ``61`` to ``1061``.
>       * :c:macro:`SO_IPV6_DELAYED_ADDR_REFRESH` - Changed from ``62`` to ``1062``.

Reviewer 24: ```suggestion
* Updated the application to use the :ref:`lib_downloader` library instead of the deprecated :ref:`lib_download_client` library.
```

Reverting the numerical values in other PR's.

Author: Updated

  * The :ref:`ug_bt_fast_pair_gatt_service_fmdn_info_callbacks_provisioning_state` section with changes to the FMDN API elements that are used for tracking of the FMDN provisioning state.

Memfault integration
--------------------

* Added a new feature to automatically post coredumps to Memfault when network connectivity is available.
  To enable this feature, set the :kconfig:option:`CONFIG_MEMFAULT_NCS_POST_COREDUMP_ON_NETWORK_CONNECTED` Kconfig option to ``y``.
  Only supported for nRF91 Series devices.

* Added a new feature to automatically capture and upload modem traces to Memfault with coredumps upon a crash.
  To enable this feature, set the :kconfig:option:`CONFIG_MEMFAULT_NCS_POST_MODEM_TRACE_ON_COREDUMP` Kconfig option to ``y``.
  Only supported for nRF91 Series devices.

sdk-nrfxlib
-----------

See the changelog for each library in the :doc:`nrfxlib documentation <nrfxlib:README>` for additional information.

* Removed the Zboss documentation.
  It is now available in separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories (depending on the device you are working with).

MCUboot
=======

> The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``a2bc982b3379d51fefda3e17a6a067342dce1a8b``, with some |NCS| specific additions.

Author: @nvlsianpu could you check the commit and the overall section?

Reviewer 18: proper hash is ``1b2fc096d9a683a7481b13749d01ca8fa78e7afd``

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

* Fixed an issue where an unusable secondary slot was cleared three times instead of once during cleanup.
* Added keys revocation scheme support for nRF54L SoCs, see ``CONFIG_BOOT_KEYS_REVOCATION`` MCUboot Kconfig option.
* Improved time performance of firmware update by usage of best RRAM write operation buffering on nRF54L SoCs.
* Introduced improved swap algorithm: swap-using-offset, see ``CONFIG_BOOT_SWAP_USING_OFFSET`` MCUboot Kconfig option.
* Image compression support has been brought to production quality.

Zephyr
======

> 

Reviewer 25: Zephyr SHAs are correct.

* :kconfig:option:`CONFIG_SETTINGS_ZMS_LL_CACHE_SIZE` - The size of the linked list cache (each entry occupies 8 Bytes of RAM).
* :kconfig:option:`CONFIG_SETTINGS_ZMS_NO_LL_DELETE` - Disables deleting the linked list nodes when deleting a Settings Key.
  Use this option only when the application is always using the same Settings Keys.
  When the application uses random Keys, enabling this option could lead to incrementing the linked list nodes without corresponding Keys and cause excessive delays to loading of the Keys.
  Use this option only to accelerate the delete operation for a fixed set of Settings elements.
* :kconfig:option:`CONFIG_SETTINGS_ZMS_LOAD_SUBTREE_PATH` - Loads first the subtree path passed in the argument, then continue to load all the Keys in the same subtree if the handler returns a zero value.
* Using :c:func:`settings_load_one` function to retrieve the value of a known Key is recommended with ZMS as it improves the read performance a lot.
* To get the value length of a Settings entry, it is recommended to use the :c:func:`settings_get_val_len` function.

zcbor

> =====

Author: @oyvindronningstad, anything to add here?

Reviewer 26: No nothing new this release.

Author: Removed the section

|no_changes_yet_note|

Trusted Firmware-M

> ==================

Reviewer 27: TF-M now supports the "Attestation service" on nRF54L15.
The TF-M documentation in NCS docs (Security/Trusted Firmware-M) is also extensively rewritten and extended. Maybe this is worth pointing out in the release notes

|no_changes_yet_note|

cJSON
=====

|no_changes_yet_note|

Documentation
=============

* Added:

>   * New page :ref:`ug_custom_board`.
>     This page includes the following subpages:

Reviewer 23: ```suggestion
  * Added an extensive documentation on :ref:`Developing with coprocessors <coprocessors_index>`.
    It covers topics related to :ref:`High-Performance Framework (HPF) <hpf_index>`, detailing how to create and integrate software peripherals using coprocessors.
  * New page :ref:`ug_custom_board`.
    This page includes the following subpages:
```

Author: Added.

    * Broadcast Source (broadcasters/Auracasters).
    * Unicast Server Source (microphones).

* :ref:`nrf_desktop`:

  * Support for Bluetooth LE legacy pairing is no longer enabled by default, because it is not secure.
  * Enabled Link Time Optimization (LTO) for images built by :ref:`Sysbuild (System build) <sysbuild>`.

Removed:

* Hardware model v1, which was deprecated in |NCS| 2.7.0, has now been removed.
  Existing projects must :ref:`transition to hardware model v2 <hw_model_v2>`.
* Multi-image builds functionality (parent-child images), which was deprecated in |NCS| v2.7.0 has now been removed.
  Existing projects must transition to :ref:`Sysbuild (System build) <sysbuild>`.
* Zigbee R22, which was deprecated in |NCS| 2.8.0, has now been removed.
  Support for `Zigbee R22`_ and `Zigbee R23`_ is available as an `nRF Connect SDK Add-on <nRF Connect SDK Add-ons_>`_.
* Asset Tracker v2 application is now removed.
  The application is replaced by `Asset Tracker Template`_, which will be available as an `nRF Connect SDK Add-on <nRF Connect SDK Add-ons_>`_.
* The application configurations for the nRF52810 Desktop Mouse board in :ref:`nrf_desktop` has been removed.

> * Sidewalk has been removed from |NCS| and is now available as an `Add-on <Amazon Sidewalk documentation_>`_.

Reviewer 23: ```suggestion
* Amazon Sidewalk has been removed from |NCS| and is now available as an `Add-on <Amazon Sidewalk documentation_>`_.
```

    * :ref:`bluetooth_mesh_light_switch`
    * :ref:`bluetooth_mesh_silvair_enocean`
    * :ref:`bluetooth_mesh_light_dim`
    * :ref:`bluetooth_mesh_light`
    * :ref:`ble_mesh_dfu_target`
    * :ref:`bluetooth_mesh_light_lc`
    * :ref:`ble_mesh_dfu_distributor`

  * Support for nRF54L05 in the following samples:

    * :ref:`bluetooth_mesh_sensor_client`
    * :ref:`bluetooth_mesh_sensor_server`
    * :ref:`bluetooth_ble_peripheral_lbs_coex`
    * :ref:`bt_mesh_chat`
    * :ref:`bluetooth_mesh_light_switch`
    * :ref:`bluetooth_mesh_silvair_enocean`
    * :ref:`bluetooth_mesh_light_dim`
    * :ref:`bluetooth_mesh_light`
    * :ref:`bluetooth_mesh_light_lc`

* Updated:

  * The board configuration files for nRF54L15, nRF54L10, and nRF54L05 by increasing the values of :kconfig:option:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` and :kconfig:option:`CONFIG_BT_RX_STACK_SIZE` Kconfig options to prevent stack overflows when :kconfig:option:`CONFIG_BT_MESH_USES_MBEDTLS_PSA` is selected.
  * The following samples to include the value of the :kconfig:option:`CONFIG_BT_COMPANY_ID` option in the Firmware ID:

    * :ref:`ble_mesh_dfu_distributor`
    * :ref:`ble_mesh_dfu_target`

>   * By disabling the friend feature when the :ref:`bluetooth_mesh_light_lc` sample is compiled for the :ref:`zephyr:nrf52dk_nrf52832` board target to increase the amount of RAM available for the application.

Reviewer 28: ```suggestion
  * :ref:`bluetooth_mesh_light_lc` sample by disabling the friend feature when the  sample is compiled for the :ref:`zephyr:nrf52dk_nrf52832` board target to increase the amount of RAM available for the application.
```

    * :ref:`npm2100_one_button` sample, demonstrating how to support wake-up, shutdown, and user interactions through a single button connected to the nPM2100.

* nRF54L Series:

  * nRF54L10 and nRF54L05 are added as supported targets in the :ref:`nrf_desktop` application.
  * :ref:`mcuboot_image_compression` is now supported on nRF54L15 and nRF54L10.
  * nRF21540 GPIO support on nRF54L Series.

Added the following features as experimental:

* nRF54L Series:

  * Bootloader and Device Firmware Update (DFU):

    * Support for nRF Secure Immutable Bootloader as first stage immutable bootloader.
    * Support for encrypted DFU with ECIES x25519 encryption using MCUboot.

  * QSPI external memory interface provided by :ref:`sQSPI Soft Peripheral <sQSPI>`, which utilizes the nRF54L15 FLPR coprocessor.
  * Coprocessor High Performance Framework, a framework designed to facilitate the creation and integration of :ref:`software peripherals using the nRF54L15 FLPR coprocessor <coprocessors_index>`.

Improved:

* Wi-Fi®:

>   * Up to 30 kB reduction in the RAM footprint of the Wi-Fi stack on nRF5340 and nRF54L15 hosts, for Wi-Fi applications with low throughput requirements.

Reviewer 29: Please change 30 kB to 25kB.

Author: Updated.

  * The following documentation pages:

    * :ref:`ug_tfm_architecture`.
    * :ref:`ug_psa_certified_api_overview`.
    * :ref:`ug_tfm_supported_services`.

* Updated:

  * The Oberon PSA core to version 1.3.4 that introduces support for the following:

    * PSA static key slots with the option :kconfig:option:`CONFIG_MBEDTLS_PSA_STATIC_KEY_SLOTS`.
    * NIST SP 800-108 conformant CMAC and HMAC based key derivation using Oberon PSA driver.

      For more information regarding the Oberon PSA core v1.3.4 update, see the relevant changelog entry in the `Oberon PSA core changelog`_.

    * The :ref:`app_approtect` page with nRF Util commands that replaced the nrfjprog commands.
      This is part of the ongoing work of archiving `nRF Command Line Tools`_ and replacing them with nRF Util.
    * The :ref:`app_boards_spe_nspe` documentation page from the :ref:`ug_app_dev` section has been moved under :ref:`ug_tfm_index`.

* Removed the Running applications with Trusted Firmware-M page.
  Its contents have been moved into the following new pages:

  * :ref:`ug_tfm_index`
  * :ref:`ug_tfm_building`
  * :ref:`ug_tfm_logging`
  * :ref:`ug_tfm_services`
  * :ref:`ug_tfm_provisioning`

> 

Reviewer 18: Please add:

west ncs-provision command was ported onto newest nrfutil device provisioning command. User must update `nrfutil; device` to version recommended for this NCS release.

Author: Added under Build and Configuration.

  * Onboarding Payload in NFC tags.
  * Large messages over TCP.
  * New ``kFactoryReset`` event that is posted during a factory reset.
    The application can register a handler and perform additional cleanup

Thread
------

* Added:

  * Support for storing the Thread key materials in the :ref:`ug_ot_thread_security_kmu`.
  * The :ref:`ug_ot_thread_security` user guide describing the security features of the OpenThread implementation in the |NCS|.

Zigbee
------

* Removed all Zigbee resources.
  They are now available as separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories.

Wi-Fi
-----

* Updated:

  * Throughputs for Wi-Fi usage profiles.
  * The Wi-Fi credential shell, by renaming it from ``wifi_cred`` to ``wifi cred``.
  * The :ref:`ug_wifi_regulatory_certification` documentation by moving it to the :ref:`ug_wifi` protocol page.

Applications
============

* Added the new :ref:`hpf_mspi_example` application.
* Removed the Asset Tracker v2 application.
  For the development of asset tracking applications, refer to the `Asset Tracker Template <Asset Tracker Template_>`_.

>   The factory-programmed Asset Tracker v2 firmware is still available to program the nRF91 Series devices using the `Programmer app`_, `Quick Start app`_, and the `Cellular Monitor app`_.

Reviewer 15: ```suggestion
  The factory-programmed Asset Tracker v2 firmware is still available to program the nRF91xx DKs using the `Programmer app`_, `Quick Start app`_, and the `Cellular Monitor app`_.
```

    * :ref:`peripheral_gatt_dm`
    * :ref:`peripheral_mds`
    * :ref:`peripheral_cts_client`

* :ref:`direct_test_mode` sample:

  * Added:

    * Loading of radio trims and a fix of a hardware errata for the nRF54H20 SoC to improve the RF performance.
    * Support for the :ref:`nRF2220 front-end module <ug_radio_fem_nrf2220ek>`.
    * Workaround for the hardware errata HMPAN-216 for the nRF54H20 SoC.

* :ref:`central_uart` sample:

  * Added reconnection to bonded devices based on their address.

* :ref:`peripheral_hids_keyboard` sample:

  * Fixed the issue with the :kconfig:option:`CONFIG_NFC_OOB_PAIRING` Kconfig option that is defined at the sample level and could not be enabled due to the unmet dependency on the :kconfig:option:`CONFIG_HAS_HW_NRF_NFCT` Kconfig option.
    The issue is resolved by enabling the ``nfct`` node in the sample devicetree configuration, which sets the :kconfig:option:`CONFIG_HAS_HW_NRF_NFCT` Kconfig option to the expected value.

* :ref:`power_profiling` sample:

  * Added the :kconfig:option:`CONFIG_BT_POWER_PROFILING_NFC_DISABLED` Kconfig option to reduce power consumption by disabling the NFC.

> 

Reviewer 30: Please mention removing of support for the nRF52805/810/811 devices in peripheral_uart sample (https://github.com/nrfconnect/sdk-nrf/commit/1f5e04d3e605aa97399700b9f30ca8d85a8cf1cb):

_* :ref:`peripheral_uart` sample: removed support for the following devices: nRF52805, nRF52810 and nRF52811_

Author: Added

* Added :ref:`soft_peripherals`.
* Removed the Zboss documentation.
  It is now available in separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories (depending on the device you are working with).

MCUboot
=======

The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``a2bc982b3379d51fefda3e17a6a067342dce1a8b``, with some |NCS| specific additions.

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

* Fixed an issue where an unusable secondary slot was cleared three times instead of once during cleanup.
* Added keys revocation scheme support for nRF54L SoCs, see ``CONFIG_BOOT_KEYS_REVOCATION`` MCUboot Kconfig option.
* Improved time performance of firmware update by usage of best RRAM write operation buffering on nRF54L SoCs.
* Introduced improved swap algorithm: swap-using-offset, see ``CONFIG_BOOT_SWAP_USING_OFFSET`` MCUboot Kconfig option.
* Image compression support has been brought to production quality.

> 

Reviewer 18: Experimental support for encrypted DFU with ECIES x25519 encryption using MCUboot for nRF54l.

Author: Added

  * :ref:`nRF21540 Front-End Module in GPIO mode <ug_radio_fem_nrf21540_gpio>` for the nRF54L Series devices.

* Fixed an issue for the nRF21540 Front-End Module (for GPIO and GPIO+SPI modes) when spurious emission occurred due to late activation of the ``TX_EN`` pin.
  The ``PDN`` pin is now activated earlier by the call to the :c:func:`mpsl_fem_enable` function.
  The ``TX_EN`` pin is now activated 15 µs earlier during the ramp-up of the radio.

Developing with PMICs
=====================

* Added the :ref:`ug_npm2100_developing` documentation.

Security
========

* Added:

  * Support for HKDF-Expand and HKDF-Extract in CRACEN.
  * Support for HashEdDSA (ed25519ph) to CRACEN.
  * TF-M now supports Attestation service on nRF54L15
  * The following documentation pages:

    * :ref:`ug_tfm_architecture`.
    * :ref:`ug_psa_certified_api_overview`.
    * :ref:`ug_tfm_supported_services`.

* Updated:

  * The Oberon PSA core to version 1.3.4 that introduces support for the following:

    * PSA static key slots with the option :kconfig:option:`CONFIG_MBEDTLS_PSA_STATIC_KEY_SLOTS`.
    * Key Wrap with and without padding (NIST-SP-800-38F) using Oberon PSA driver.

>     * WPA3-SAE and WPA3-SAE-PT using Oberon PSA driver.

Reviewer 27: Let's remove lines mentioning "Key wrap" and "WPA3-SAE".
Those are not standardized as PSA Crypto features yet, and they should not be relied on for external users of the SDK. They are only available for an internal evaluation so far.

Author: Removed

Zigbee
------

* Removed all Zigbee resources.
  They are now available as separate `Zigbee R22`_ and `Zigbee R23`_ add-on repositories.

Wi-Fi
-----

* Updated:

  * Throughputs for Wi-Fi usage profiles.
  * The Wi-Fi credential shell, by renaming it from ``wifi_cred`` to ``wifi cred``.
  * The :ref:`ug_wifi_regulatory_certification` documentation by moving it to the :ref:`ug_wifi` protocol page.

Applications
============

* Added the new :ref:`hpf_mspi_example` application.
* Removed the Asset Tracker v2 application.
  For the development of asset tracking applications, refer to the `Asset Tracker Template <Asset Tracker Template_>`_.

  The factory-programmed Asset Tracker v2 firmware is still available to program the nRF91 Series devices using the `Programmer app`_, `Quick Start app`_, and the `Cellular Monitor app`_.

* Renamed the SDP GPIO application to :ref:`hpf_gpio_example`.

Machine learning
----------------

* Updated:

  * The application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target.
  * The Edge Impulse URI configuration to use the new model location.

> IPC radio firmware

Reviewer 30: If we stick to alphabetical order of the subsections - please move _IPC radio firmware_ above _Machine learning_.

> ------------------

Reviewer 30: If we stick to alphabetical order of the subsections - please move _IPC radio firmware_ above _Machine learning_.

Author: Moved.

* Updated:

  * The application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system in all devices that contain MRAM, such as the nRF54H Series devices.
  * The documentation of applications and samples that use the IPC radio firmware as a :ref:`companion component <companion_components>` to mention its usage when built with :ref:`configuration_system_overview_sysbuild`.

> 

Reviewer 30: I suggest to add a note about a fix:

_* Fixed a performance issue where the :ref:`ipc_radio` application could drop HCI packets in case of high data traffic_

Author: Added.

    It is needed for the radio protocols to operate correctly.
  * MRAM always-on request for scheduled radio events.
    It is needed to avoid MRAM wake-up latency for radio protocols.

Libraries for networking
------------------------

* Added:

  * The :ref:`lib_downloader` library.
  * A backend for the :ref:`TLS Credentials Subsystem <zephyr:sockets_tls_credentials_subsys>` that stores the credentials in the modem, see :kconfig:option:`CONFIG_TLS_CREDENTIALS_BACKEND_NRF_MODEM`.

* Deprecated the :ref:`lib_download_client` library.
  See the :ref:`migration guide <migration_3.0_recommended>` for recommended changes.

* Updated the following libraries to use the :ref:`lib_downloader` library instead of the :ref:`lib_download_client` library:

  * :ref:`lib_nrf_cloud`
  * :ref:`lib_aws_fota`
  * :ref:`lib_azure_fota`
  * :ref:`lib_fota_download`

* :ref:`lib_nrf_cloud_pgps` library:

  * Fixed the warning due to missing ``https`` download protocol.

* :ref:`lib_fota_download` library:

  * Added error codes related to unsupported protocol, DFU failures, and invalid configuration.

>   * Updated to use the :ref:`lib_downloader` library for CoAP downloads.

Reviewer 31: this line is redundant, already mentioned in 1253

Author: Removed the redundant entry.

* :ref:`lib_nrf_cloud` library:

  * Added the :kconfig:option:`CONFIG_NRF_CLOUD` Kconfig option to prevent unintended inclusion of nRF Cloud Kconfig variables in non-nRF Cloud projects.
  * Updated:

>     * To use the :ref:`lib_downloader` library for CoAP downloads.

Reviewer 31: this line is redundant with the above mention

Author: Removed the redundant entry.

.. note::
   For possible breaking changes and changes between the latest Zephyr release and the current Zephyr version, refer to the :ref:`Zephyr release notes <zephyr_release_notes>`.

Additions specific to |NCS|
---------------------------

The new ZMS Settings backend is not backward compatible with the old version.

Using the new backend you can now enable some performance optimizations using the following Kconfig options:

* :kconfig:option:`CONFIG_SETTINGS_ZMS_LL_CACHE` - Used for caching the linked list nodes related to Settings Key/Value entries.
* :kconfig:option:`CONFIG_SETTINGS_ZMS_LL_CACHE_SIZE` - The size of the linked list cache (each entry occupies 8 Bytes of RAM).
* :kconfig:option:`CONFIG_SETTINGS_ZMS_NO_LL_DELETE` - Disables deleting the linked list nodes when deleting a Settings Key.
  Use this option only when the application is always using the same Settings Keys.
  When the application uses random Keys, enabling this option could lead to incrementing the linked list nodes without corresponding Keys and cause excessive delays to loading of the Keys.
  Use this option only to accelerate the delete operation for a fixed set of Settings elements.

> * :kconfig:option:`CONFIG_SETTINGS_ZMS_LOAD_SUBTREE_PATH` - Loads first the subtree path passed in the argument, then continue to load all the Keys in the same subtree if the handler returns a zero value.

Reviewer 7: ```suggestion
* :kconfig:option:`CONFIG_SETTINGS_ZMS_LOAD_SUBTREE_PATH` - First loads the subtree path passed in the argument, then continues to load all the Keys in the same subtree if the handler returns a zero value.
```

> * Using :c:func:`settings_load_one` function to retrieve the value of a known Key is recommended with ZMS as it improves the read performance a lot.

Reviewer 7: ```suggestion
* Using the :c:func:`settings_load_one` function to retrieve the value of a known Key is recommended with ZMS as it improves the read performance a lot.
```

* To get the value length of a Settings entry, it is recommended to use the :c:func:`settings_get_val_len` function.

zcbor
=====

|no_changes_yet_note|

Documentation
=============

* Added:

  * Extensive documentation on :ref:`Developing with coprocessors <coprocessors_index>`.
    It covers topics related to :ref:`High-Performance Framework (HPF) <hpf_index>`, detailing how to create and integrate software peripherals using coprocessors.
  * New page :ref:`ug_custom_board`.
    This page includes the following subpages:

    * :ref:`defining_custom_board` - Previously located under :ref:`app_boards`.
    * :ref:`programming_custom_board` - New subpage.

  * New page :ref:`thingy53_precompiled` under :ref:`ug_thingy53`.
    This page includes some of the information previously located on the standalone page for Getting started with Nordic Thingy:53.
  * New page :ref:`add_new_led_example` under :ref:`configuring_devicetree`.
    This page includes information previously located in the |nRFVSC| documentation.

>   * The :ref:`cellular_psm` page under the :ref:`ug_lte` documentation, and the documentation is now spilt into subpages.

Reviewer 7: ```suggestion
  * The :ref:`cellular_psm` page under the :ref:`ug_lte` documentation, and the documentation is now split into subpages.
```

  * Bluetooth controller and host qualified for the nRF54H20.

* Matter:

  * Matter 1.4.1: Matter commissioning using NFC tag can be officially certified now.
  * :ref:`matter_manufacturer_specific_sample` sample: Dedicated sample, containing documentation and a preview :ref:`PC tool <ug_nrfconnect_manufacturer_cluster_editor_tool>`, facilitating creation and modification of manufacturer specific clusters.

* PMIC:

  * nPM2100 and nPM2100 EK:

    * Added support for nPM2100, which is a PMIC designed for primary (non-rechargeable) batteries in an extremely compact form factor.
      It has an ultra-efficient boost regulator, a dual purpose LDO/load switch, two GPIOs, an ADC, and other features.
    * :ref:`npm2100_fuel_gauge` sample, demonstrating how to calculate the state of charge of a supported primary cell battery using the nPM2100 and the :ref:`nrfxlib:nrf_fuel_gauge`.
    * :ref:`npm2100_one_button` sample, demonstrating how to support wake-up, shutdown, and user interactions through a single button connected to the nPM2100.

* nRF54L Series:

  * Bootloader and Device Firmware Update (DFU):

    * nRF Secure Immutable Bootloader. (to be updated)

>     * MCUboot encrypted DFU using ECIES.(to be updated)

Reviewer 6: Having these features as supported seems to contradict the SML tables:

https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/software_maturity.html#nrf_secure_immutable_bootloader

https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/software_maturity.html#mcuboot_bootloader

CC: @nvlsianpu

Author: These are now moved to experimental (as agreed with PMT)
